# Data Distillery Data Dictionary


[TOC]





# Introduction

The Data Distillery project aims to integrate summarized (“distilled”) Common Fund data within a knowledge graph. The purpose of the Data Distillery Knowledge Graph (DDKG) is to link multiple sources of expertly curated data, thus providing data integration across multiple Common Fund data coordinating centers (DCCs).  The summarized data are provided by participating DCCs and funded as part of the Common Fund Data Ecosystem (CFDE) project. The DDKG schema is based on the Unified Biomedical Knowledge Graph (UBKG) which originates from the Unifield Medical Language System (UMLS). The UBKG supports the DDKG with over 180 different ontologies and standards supporting the Common Fund data that either are native to UMLS or were explicitly added to support biomolecular data (see Figure 1).  The DDKG can be used to create simple to complex queries, and use the results for a range of different applications related to the use of Common Fund data. We include some use cases in another document &lt;link to Use Case Document here>.

For the first phase of the project, the participating DCCs have submitted 29 different datasets for integration into the DDKG This document is focused on outlining these 29 datasets within the DDKG,, and describing the schema and information on each dataset.

Figure 1: Base datasets


# Base Datasets

Information on the base set of ontologies included in the Data Distillery Knowledge Graph can be found in the [documentation for the Unified Biomedical Knowledge Graph (UBKG)](https://ubkg.docs.xconsortia.org/contexts/#ubkg-base-context-umls-plus), upon which the Data Distillery is built. See Figure 1 for a general schematic.


# <span style="text-decoration:underline;">DCC Datasets</span>


## 4D Nucleome (4DN) DCC


### 4D Nucleome datasets


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>4DNQ 4DNL 4DNF 4DND
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td>data.4dnucleome.org
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>4DN-DCIC
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Andy Schroeder (PM) Harvard Medical School, Boston
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>Chromatin loops called from Hi-C experiments performed in select cell lines.  
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>Representing topologically associated domains and loops by chromosomal location can allow exploration of gene expression, genomic variation and other biological information in the context of chromatin architecture. 
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>Hi-C chromatin capture assays generate information on regions of the genome that can be located far apart along the linear sequence of DNA but are in close physical proximity in nuclear chromatin. Architectural features of the chromatin including topologically associated domains (TADs), loops and dots can be generated by algorithms from the results of Hi-C experiments.
<p>
Loop calls from several 4DNucleome Hi-C datasets generated from select cell lines and tissues are provided to the data distillery for ingestion.
<p>
A subset of loop calls were generated by two different 4DN research labs on datasets from the H1-ESC human ES cell line, H1 differentiated to endoderm and HFFc6, a human foreskin derived cell line.  Loop calls from the Dekker lab were generated using the cooltools re-implementation of HICCUPS as described in Oksuz et al. 2021 https://pubmed.ncbi.nlm.nih.gov/34480151/.  Loop calls from the Cremins lab were generated as described in Emerson et al. 2022 https://pubmed.ncbi.nlm.nih.gov/35676475/.  Additional calls from the Cremins lab generated and part of the data from the Emerson paper from the HCT116 colorectal cancer cell line with or without depletion of WAPL or RAD21, genes that encode protein important for chromatin architecture are also provided.
<p>
In addition, the 4DN-DCIC generated loop calls on 4 additional datasets, in situ Hi-C performed on H1-ESC or GM12878 cell lines, 4DNESFSCP5L8 and 4DNES3JX38V5, respectively and DNase-Hi-C in fetal heart tissue (4DNESZFHB53P) or RUES2 stem cells differentiated to cardiomyocytes (4DNESGTHHJAC). These loop calls from the 25 kb resolution matrices of these datasets were further filtered for those loops that overlapped expressed genes identified from gene expression data from the same or comparable cells and tissues.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td><a href="https://docs.google.com/spreadsheets/d/1dj51N-MYRQ_CYPPDp_zAX0thJcfXcsFBDi_jxYAcRq0/edit?usp=sharing">This document</a> indicates the datasets and files used as input to the 4DN distilled data.  For the genome-wide loop calls from the Dekker and Cremins groups the indicated files were the direct input into the summarization process.  For the 4DN-DCIC loop calls the mcool files indicated were used to call loops and further summarized utilizing expression data to provide a file of loops that overlap expressed genes as described in <a href="https://docs.google.com/document/d/1GeTbwshqr2HD1JIb8z33Z_3nspTVTJdyRoUmQl6o9Xo/edit?usp=sharing">this document</a>.
<p>
Provided loop files were further prepared for ingestion by first creating dataset nodes (SAB: ‘<strong>4DND</strong>’) with the respective terms containing the dataset information (assay type, lab and cell type involved), file nodes (SAB: ‘<strong>4DNF</strong>’) with the respective terms containing the file information, loop nodes (SAB: ‘<strong>4DNL</strong>’) attached to HSCLO nodes at 1kpb resolution level corresponding to upstream start and end and downstream start and end nodes of the characteristic anchor of the loop and q-value nodes (SAB: ‘<strong>4DNQ</strong>’) corresponding to donut q-value of the loops. The mentioned nodes are then used to create concept nodes with connections depicted in the schematic below.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td><a href="https://github.com/TaylorResearchLab/CFDE_DataDistillery/tree/main/DCC_workflows/4DN">https://github.com/TaylorResearchLab/CFDE_DataDistillery/tree/main/DCC_workflows/4DN</a>
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>462,178
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>2,768,612
   </td>
  </tr>
</table>





### 4DN Schema Diagram



<p id="gdcalert1" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image1.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert2">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image1.png "image_tooltip")



###  \
4DN Node and Edge counts


<table>
  <tr>
   <td><strong>SAB</strong>
   </td>
   <td><strong>Count</strong>
   </td>
  </tr>
  <tr>
   <td><strong>4DND</strong>
   </td>
   <td>12
   </td>
  </tr>
  <tr>
   <td><strong>4DNF</strong>
   </td>
   <td>12
   </td>
  </tr>
  <tr>
   <td><strong>4DNL</strong>
   </td>
   <td>215822
   </td>
  </tr>
  <tr>
   <td><strong>4DNQ</strong>
   </td>
   <td>354
   </td>
  </tr>
</table>



<table>
  <tr>
   <td><strong>Subject SAB</strong>
   </td>
   <td><strong>Predicate</strong>
   </td>
   <td><strong>Object SAB</strong>
   </td>
   <td><strong>Count</strong>
   </td>
  </tr>
  <tr>
   <td><strong>4DND</strong>
   </td>
   <td>has_assay_type
   </td>
   <td>EFO
   </td>
   <td>12
   </td>
  </tr>
  <tr>
   <td><strong>4DND</strong>
   </td>
   <td>has_assay_type
   </td>
   <td>OBI
   </td>
   <td>12
   </td>
  </tr>
  <tr>
   <td><strong>4DND</strong>
   </td>
   <td>dataset_involved_cell_type
   </td>
   <td>EFO
   </td>
   <td>11
   </td>
  </tr>
  <tr>
   <td><strong>4DND</strong>
   </td>
   <td>dataset_involved_cell_type
   </td>
   <td>UBERON
   </td>
   <td>1
   </td>
  </tr>
  <tr>
   <td><strong>4DND</strong>
   </td>
   <td>dataset_has_file
   </td>
   <td>4DNF
   </td>
   <td>12
   </td>
  </tr>
  <tr>
   <td><strong>4DNF</strong>
   </td>
   <td>file_has_loop
   </td>
   <td>4DNL
   </td>
   <td>215822
   </td>
  </tr>
  <tr>
   <td><strong>4DNL</strong>
   </td>
   <td>loop_has_qvalue_bin
   </td>
   <td>4DNQ
   </td>
   <td>215822
   </td>
  </tr>
  <tr>
   <td><strong>4DNL</strong>
   </td>
   <td>loop_us_start
   </td>
   <td>HSCLO
   </td>
   <td>215822
   </td>
  </tr>
  <tr>
   <td><strong>4DNL</strong>
   </td>
   <td>loop_us_end
   </td>
   <td>HSCLO
   </td>
   <td>215822
   </td>
  </tr>
  <tr>
   <td><strong>4DNL</strong>
   </td>
   <td>loop_ds_start
   </td>
   <td>HSCLO
   </td>
   <td>215822
   </td>
  </tr>
  <tr>
   <td><strong>4DNL</strong>
   </td>
   <td>loop_ds_end
   </td>
   <td>HSCLO
   </td>
   <td>215822
   </td>
  </tr>
</table>



## <span style="text-decoration:underline;">Extracellular RNA Communication Program (ERCC)</span>


### ERCC RBP dataset


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>ENSEMBL, UBERON, UNIPROTKB, ENCODE.RBS.150.NO.OVERLAP, ENCODE.RBS.HepG2, ENCODE.RBS.HepG2.K562, ENCODE.RBS.K562
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td><a href="https://exrna.org">https://exrna.org</a> 
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>Extracellular RNA Communication Consortium (ERCC)
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Aleksandar Milosavljevic
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>The genomic coordinates of eCLIP peaks of 150 RNA binding proteins (RBPs) were taken from eCLIP-seq analysis results published by the ENCODE project. Control extracellular RNA (exRNA) sequencing profiles available through the exRNA Atlas were used to draw several relationships.
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>To help identify minimally invasive biomarkers of disease.
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>Assertions describe relationships between RBPs, RBP binding sites, genes, and biofluids. RBP binding sites refers to both eCLIP peaks and a second type of genomic locus. The group is the result of trimming the eCLIP loci so that there are no overlaps between sets of loci from a given pair of RBPs.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td>Relationships between RBPs and biofluids, and between eCLIP loci and biofluids are the result of a correlation-based analysis. This analysis was performed using the coverage of trimmed eCLIP loci within control exRNA profiles made available through the exRNA Atlas. This analysis is described in detail by LaPlante et al., Cell Genomics, 2023.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td><a href="https://github.com/TaylorResearchLab/CFDE_DataDistillery/blob/main/DCC_workflows/ERCC/check_ERCC_submissions.ipynb">https://github.com/TaylorResearchLab/CFDE_DataDistillery/blob/main/DCC_workflows/ERCC/check_ERCC_submissions.ipynb</a>
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>1,169,178
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>2,431,786
   </td>
  </tr>
  <tr>
   <td>Source Data DOI(s) <em>(optional)</em>
   </td>
   <td><a href="https://doi.org/10.1038/nmeth.3810">https://doi.org/10.1038/nmeth.3810</a> 
   </td>
  </tr>
  <tr>
   <td>Source Data URL(s) <em>(optional)</em>
   </td>
   <td><a href="https://www.encodeproject.org/encore-matrix/?type=Experiment&status=released&internal_tags=ENCORE">https://www.encodeproject.org/encore-matrix/?type=Experiment&status=released&internal_tags=ENCORE</a> 
   </td>
  </tr>
</table>



### ERCC RBP Schema Diagram 

<p id="gdcalert2" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image2.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert3">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image2.png "image_tooltip")



### ERCC RBP Node and Edge counts



<p id="gdcalert3" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image3.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert4">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image3.png "image_tooltip")


<p id="gdcalert4" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image4.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert5">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image4.png "image_tooltip")



### ERCC Regulatory Element dataset


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>CLINGEN.ALLELE.REGISTRY, ENSEMBL, GTEXEQTL, UBERON, ENCODE.CCRE, ENCODE.CCRE.ACTIVITY, ENCODE.CCRE.CTCF, ENCODE.CCRE.H3K27AC, ENCODE.CCRE.H3K4ME3 (node SABs)
<p>
<strong>ERCCREG, ERCCRBP</strong> (edge SABs)
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td><a href="https://exrna.org">https://exrna.org</a> 
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>Extracellular RNA Communication Consortium (ERCC)
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Aleksandar Milosavljevic
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>The results of CHIP-seq experiments conducted by the ENCODE project were used to identify regulatory elements active within specific tissues and their transcriptional role. Similarly, we used data published by the GTEx project to identify eQTLs active within specific tissues.
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>To identify regulatory elements active within a specific tissue which are also supported by having an active eQTL within the range of its genomic coordinates.
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>The tissue specific regulation of a gene by an eQTL is modeled using variant, tissue, eQTL, and gene nodes. The same model structure is also used for regulatory elements. In this case a “regulatory element activity” (SAB=<strong>ENCODE.CCRE.ACTIVITY</strong>) node is used as the central node rather than the eQTL node. Regulatory element activity nodes are also decorated with relationships to other nodes to assist in determining the tissue specific transcriptional role of the regulatory element. eQTL and regulatory element models are connected by a relationship between variant and regulatory element nodes.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td>To summarize regulatory element data, ENCODE biosamples were grouped by their respective tissue or cell line ontology code. These groups were then further grouped by the number of samples within each biosample group. Next, within the DNase Z-score data matrix provided by ENCODE, for each larger group and each regulatory element, the number of z-scores that were above 1.64 were counted within samples of each biosample (or small) group. This process was used to build a reference distribution of counts specific to a biosample group with a specific number of members. Regulatory elements were then classified as active within a specific tissue or cell type if the count of z-scores greater than 1.64 within ENCODE biosamples belonging to that group was above the median value of the reference distribution.
<p>
Only regulatory elements classified as active in at least one tissue or cell type are included. The process described above was repeated for the H3K4me3, H3K27Ac, and CTCF z-score data matrices to decorate regulatory element activity nodes with relationships to other nodes to help the user identify the transcriptional role of each regulatory element.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td><a href="https://github.com/TaylorResearchLab/CFDE_DataDistillery/blob/main/DCC_workflows/ERCC/check_ERCC_submissions.ipynb">https://github.com/TaylorResearchLab/CFDE_DataDistillery/blob/main/DCC_workflows/ERCC/check_ERCC_submissions.ipynb</a>
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>2,918,828
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>14,897,093
   </td>
  </tr>
  <tr>
   <td>Source Data DOI(s) <em>(optional)</em>
   </td>
   <td><a href="https://doi.org/10.1038/s41586-020-2493-4">https://doi.org/10.1038/s41586-020-2493-4</a> 
   </td>
  </tr>
  <tr>
   <td>Source Data URL(s) <em>(optional)</em>
   </td>
   <td><a href="https://screen.wenglab.org/">https://screen.wenglab.org/</a> 
<p>
<a href="https://www.gtexportal.org/home/">https://www.gtexportal.org/home/</a> 
   </td>
  </tr>
</table>



### ERCC Regulatory Element Schema Diagram



<p id="gdcalert5" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image5.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert6">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image5.png "image_tooltip")



### ERCC Regulatory Element Node and Edge Counts



<p id="gdcalert6" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image6.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert7">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image6.png "image_tooltip")




<p id="gdcalert7" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image7.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert8">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image7.png "image_tooltip")



## GlyGen


### GlyGen Datasets


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>FALDO, GLYCOCOO,GLYCORDF, UNIPROTKB, PROTEOFORM, GLACANS
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td>https://www.glygen.org/
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>GlyGen
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Raja Mazumder (PI) George Washington University; Mike Tiemeyer (PI) University of Georgia
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>Data for GlyGen is retrieved from multiple glycomics database (e.g. GlyTouCan, GlyConnect, MatrixDB), proteomics database (e.g. UniProtKB) and other domain database (e.g. Ensembl, RefSeq, BioMuta, OMA, MGI, Bgee). All data is transformed in standardized representation and integrate in GlyGen
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>Provide computational and informatics resources and tools for glycosciences research. Integrate data and knowledge from diverse disciplines relevant to glycobiology. Address needs inside and outside the glycoscience community.
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>GlyGen is a data integration and dissemination project for carbohydrate and glycoconjugate related data. GlyGen retrieves information from multiple international data sources and integrates and harmonizes this data. This web portal allows exploring this data and performing unique searches that cannot be executed in any of the integrated databases alone.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td>The data ingested for the KnowledgeGraph are from ontologies associated with glycan and proteoform domain. Select nodes and edges for glycans are retrieved from GlyCoCoo and GlyCoRDF. ontologies that describe the properties of glycans.  \
The assertion data received from GlyGen in n-triples format (glycan.nt and proteoform.nt) were imported into the No4j environment using the n10s plug-in functions. Once the data was imported for each of the glycans and proteoform datasets, subgraphs were created<code>.</code>Finally, the resulting graph nodes and edges were exported as .csv files using APOC plug-in procedures.The resulting nodes and edges were reformatted by curating the relationship names and adding SABs for all entities (either by using existing SABs e.g. <strong>UNIPROTKB</strong> and <strong>GLYTOUCAN</strong> or creating custom SABs such as <strong>GLYGEN.LOCATION</strong> or <strong>GLYCOPROTEIN</strong>) and saved as the OWLNETS_node_metadata.tsv and OWLNETS_edgelist.tsv for ingestion. More information on FALDO can be found here: <a href="https://bioportal.bioontology.org/ontologies/FALDO">https://bioportal.bioontology.org/ontologies/FALDO</a>. More information on GlycoRDF can be found here: <a href="https://github.com/glycoinfo/GlycoRDF">https://github.com/glycoinfo/GlycoRDF</a>. More information on GlycoCoO can be found here: <a href="https://github.com/glycoinfo/GlycoCoO">https://github.com/glycoinfo/GlycoCoO</a>. Data under SAB FALDO was ingested to assist in ..
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td><a href="https://github.com/TaylorResearchLab/CFDE_DataDistillery/blob/main/DCC_workflows/GLYGEN/GlyGen_workfolw.md">https://github.com/TaylorResearchLab/CFDE_DataDistillery/blob/main/DCC_workflows/GLYGEN/GlyGen_workfolw.md</a>
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>154, 241770 (PROTEOFORM), 182269 (GLYCANS)
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>137, 455469 (PROTEOFORM), 464659 (GLYCANS)
   </td>
  </tr>
  <tr>
   <td>Source Data DOI(s) <em>(optional)</em>
   </td>
   <td>Raw data DOI
   </td>
  </tr>
  <tr>
   <td>Source Data URL(s) <em>(optional)</em>
   </td>
   <td>Download from <a href="https://sparql.glygen.org/">https://sparql.glygen.org/</a>. 
<p>
Data file: <a href="https://sparql.glygen.org/ln2triplestoredata/triples.tar.gz">https://sparql.glygen.org/ln2triplestoredata/triples.tar.gz</a> 
   </td>
  </tr>
</table>



### GLYGEN Schema Diagram



<p id="gdcalert8" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image8.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert9">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image8.png "image_tooltip")


<span style="text-decoration:underline;">Node and Edge counts</span> (FALDO)



<p id="gdcalert9" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image9.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert10">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image9.png "image_tooltip")




<p id="gdcalert10" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image10.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert11">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image10.png "image_tooltip")



### GlyGen Node and Edge counts (GLYCOCOO)



<p id="gdcalert11" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image11.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert12">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image11.png "image_tooltip")


<p id="gdcalert12" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image12.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert13">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image12.png "image_tooltip")


<span style="text-decoration:underline;">Node and Edge counts</span> (GLYCORDF)



<p id="gdcalert13" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image13.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert14">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image13.png "image_tooltip")




<p id="gdcalert14" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image14.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert15">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image14.png "image_tooltip")


Node and Edge Counts (PROTEOFORM) \



<table>
  <tr>
   <td><strong>SAB</strong>
   </td>
   <td><strong>Count</strong>
   </td>
  </tr>
  <tr>
   <td><strong>UNIPROTKB</strong>
   </td>
   <td>16810
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOPROTEIN</strong>
   </td>
   <td>63441
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOPROTEIN.EVIDENCE</strong>
   </td>
   <td>2088
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOSYLATION.SITE</strong>
   </td>
   <td>52451
   </td>
  </tr>
  <tr>
   <td><strong>GLYGEN.LOCATION</strong>
   </td>
   <td>52450
   </td>
  </tr>
  <tr>
   <td><strong>UNIPROTKB.ISOFORM</strong>
   </td>
   <td>8406
   </td>
  </tr>
  <tr>
   <td><strong>GLYGEN.CITATION</strong>
   </td>
   <td>2088
   </td>
  </tr>
  <tr>
   <td><strong>GP.ID2PRO</strong>
   </td>
   <td>61353
   </td>
  </tr>
  <tr>
   <td><strong>GLYTOUCAN</strong>
   </td>
   <td>1554
   </td>
  </tr>
  <tr>
   <td><strong>AMINO.ACID</strong>
   </td>
   <td>7
   </td>
  </tr>
</table>


 \



<table>
  <tr>
   <td><strong>Subject SAB</strong>
   </td>
   <td><strong>Predicate</strong>
   </td>
   <td><strong>Object SAB</strong>
   </td>
   <td><strong>Count</strong>
   </td>
  </tr>
  <tr>
   <td><strong>UNIPROTKB</strong>
   </td>
   <td>has_isoform
   </td>
   <td>UNIPROTKB.ISOFORM
   </td>
   <td>8404
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOPROTEIN</strong>
   </td>
   <td>has_evidence
   </td>
   <td>GLYCOPROTEIN.EVIDENCE
   </td>
   <td>120158
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOPROTEIN</strong>
   </td>
   <td>sequence
   </td>
   <td>UNIPROTKB.ISOFORM
   </td>
   <td>61353
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOPROTEIN.EVIDENCE</strong>
   </td>
   <td>citation
   </td>
   <td>GLYGEN.CITATION
   </td>
   <td>2088
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOPROTEIN</strong>
   </td>
   <td>has_pro_entry
   </td>
   <td>GP.ID2PRO
   </td>
   <td>61353
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOPROTEIN</strong>
   </td>
   <td>glycosylated_at
   </td>
   <td>GLYCOSYLATION.SITE
   </td>
   <td>52450
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOSYLATION.SITE</strong>
   </td>
   <td>location
   </td>
   <td>GLYGEN.LOCATION
   </td>
   <td>52450
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOSYLATION.SITE</strong>
   </td>
   <td>has_saccharide
   </td>
   <td>GLYTOUCAN
   </td>
   <td>44763
   </td>
  </tr>
  <tr>
   <td><strong>GLYGEN.LOCATION</strong>
   </td>
   <td>has_amino_acid
   </td>
   <td>AMINO.ACID
   </td>
   <td>52450
   </td>
  </tr>
</table>


Node and Edge Counts (GLYCANS) \



<table>
  <tr>
   <td><strong>SAB</strong>
   </td>
   <td><strong>Count</strong>
   </td>
  </tr>
  <tr>
   <td><strong>GLYGEN.GLYCOSYLATION</strong>
   </td>
   <td>91
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOSYLTRANSFERASE.REACTION</strong>
   </td>
   <td>91
   </td>
  </tr>
  <tr>
   <td><strong>GLYTOUCAN</strong>
   </td>
   <td>33755
   </td>
  </tr>
  <tr>
   <td><strong>GLYGEN.RESIDUE</strong>
   </td>
   <td>80
   </td>
  </tr>
  <tr>
   <td><strong>GLYGEN.SRC</strong>
   </td>
   <td>30986
   </td>
  </tr>
  <tr>
   <td><strong>GLYGEN.GLYCOSEQUENCE</strong>
   </td>
   <td>117146
   </td>
  </tr>
  <tr>
   <td><strong>GLYCAN.MOTIF</strong>
   </td>
   <td>120
   </td>
  </tr>
</table>


 \



<table>
  <tr>
   <td><strong>Subject SAB</strong>
   </td>
   <td><strong>Predicate</strong>
   </td>
   <td><strong>Object SAB</strong>
   </td>
   <td><strong>Count</strong>
   </td>
  </tr>
  <tr>
   <td><strong>GLYGEN.GLYCOSYLATION</strong>
   </td>
   <td>has_enzyme_protein
   </td>
   <td>UNIPROTKB
   </td>
   <td>91
   </td>
  </tr>
  <tr>
   <td><strong>GLYCOSYLTRANSFERASE.REACTION</strong>
   </td>
   <td>has_enzyme_protein
   </td>
   <td>UNIPROTKB
   </td>
   <td>91
   </td>
  </tr>
  <tr>
   <td><strong>GLYTOUCAN</strong>
   </td>
   <td>is_from_source
   </td>
   <td>GLYGEN.SRC
   </td>
   <td>30986
   </td>
  </tr>
  <tr>
   <td><strong>GLYTOUCAN</strong>
   </td>
   <td>has_glycosequence
   </td>
   <td>GLYGEN.GLYCOSEQUENCE
   </td>
   <td>117146
   </td>
  </tr>
  <tr>
   <td><strong>GLYGEN.RESIDUE</strong>
   </td>
   <td>attached_by
   </td>
   <td>GLYGEN.GLYCOSYLATION
   </td>
   <td>349
   </td>
  </tr>
  <tr>
   <td><strong>GLYTOUCAN</strong>
   </td>
   <td>synthesized_by
   </td>
   <td>GLYCOSYLTRANSFERASE.REACTION
   </td>
   <td>210563
   </td>
  </tr>
  <tr>
   <td><strong>GLYTOUCAN</strong>
   </td>
   <td>has_motif
   </td>
   <td>GLYCAN.MOTIF
   </td>
   <td>19321
   </td>
  </tr>
  <tr>
   <td><strong>GLYTOUCAN</strong>
   </td>
   <td>has_canonical_residue
   </td>
   <td>GLYGEN.RESIDUE
   </td>
   <td>86033
   </td>
  </tr>
  <tr>
   <td><strong>GLYGEN.RESIDUE</strong>
   </td>
   <td>has_parent
   </td>
   <td>GLYGEN.RESIDUE
   </td>
   <td>79
   </td>
  </tr>
</table>



## Genotype Tissue Expression (GTEx)


### GTEx datasets


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>GTEXEXP, GTEXEQTL, EXPBINS, PVALUEBINS
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td><a href="https://www.gtexportal.org/home/">https://www.gtexportal.org/home/</a>
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>GTEx
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Kristen Ardlie 
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>Documentation on the sources of GTEx data can be found here: <a href="https://biospecimens.cancer.gov/resources/sops/docs/GTEx_SOPs/BBRB-PR-0004-W1%20GTEx%20Tissue%20Harvesting%20Work%20Instruction.pdf">https://biospecimens.cancer.gov/resources/sops/docs/GTEx_SOPs/BBRB-PR-0004-W1%20GTEx%20Tissue%20Harvesting%20Work%20Instruction.pdf</a>
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>To include bulk RNA-seq gene expression levels from adult tissues as well as correlations between genotype and tissue-specific gene expression levels as expression quantitative trait loci (eQTLs) that identify regions of the genome that influence whether and how much a gene is expressed. 
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>The Genotype-Tissue Expression (GTEx) project is an ongoing effort to build a comprehensive public resource to study tissue-specific gene expression and regulation. Samples were collected from 54 non-diseased tissue sites across nearly 1000 individuals, primarily for molecular assays including WGS, WES, and RNA-Seq. This database includes expression levels for genes by tissue in terms of transcripts per mission (TPM). The database also contains the p-values and relationships between loci and genes as expression quantitative trait loci (eQTLs). 
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td>Three types of GTEx data were summarized and ingested into the knowledge graph listed below by SAB:
<p>
(1) <strong>GTEXEXP</strong> -  Transcript per million (TPM) values, which represent gene-tissue expression levels, were ingested as is except that edges to ‘bin nodes’ (EXPBINS) were created.
<p>
For example, a GTEXEXP node with a TPM of 10.5 will have an edge to the bin node that represents [10,11] TPM.
<p>
(2) <strong>GTEXEQTL</strong> - GTEx eQTLs were filtered to include only those that are present in every tissue. This reduced the total set of eQTLs to ~2 million. P-values for the eQTLs are also included in the graph, however, they are represented as bin nodes just like the TPM values for the GTEXEXP dataset.
<p>
(3) <strong>GTEXCOEXP</strong> - A GTEx co-expression dataset was made by first calculating the Pearson’s correlation coefficient of all genes in GTEx intersection with HGNC master list separately for each of 54 tissues listed in GTEx using the provided TPMs. Then pairs of genes with correlation coefficient > 0.99 were tagged in each tissue as strongly correlated and reported as assertions with relationship types and counts in the attached table (please see below)
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td><a href="https://github.com/TaylorResearchLab/CFDE_DataDistillery/tree/main/DCC_workflows/GTEx">https://github.com/TaylorResearchLab/CFDE_DataDistillery/tree/main/DCC_workflows/GTEx</a>
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>6,280,011
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>31,904,034
   </td>
  </tr>
  <tr>
   <td>Source Data DOI(s) <em>(optional)</em>
   </td>
   <td>Raw data DOI
   </td>
  </tr>
  <tr>
   <td>Source Data URL(s) <em>(optional)</em>
   </td>
   <td><a href="https://www.gtexportal.org/home/datasets">https://www.gtexportal.org/home/datasets</a>
<p>
(GTEx_Analysis_v8_eQTL.tar
<p>
and
<p>
GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct.gz)
   </td>
  </tr>
</table>



### GTEXEXP Schema Diagram



<p id="gdcalert15" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image15.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert16">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image15.png "image_tooltip")



### GTEXQTL Schema Diagram



<p id="gdcalert16" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image16.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert17">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image16.png "image_tooltip")



### GTEx Node and Edge counts (GTEXEXP/EXPBINS)



<p id="gdcalert17" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image17.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert18">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image17.png "image_tooltip")




<p id="gdcalert18" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image18.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert19">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image18.png "image_tooltip")
 

<span style="text-decoration:underline;">Node and Edge counts</span> (GTEXEQTL/PVALUEBINS)



<p id="gdcalert19" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image19.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert20">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image19.png "image_tooltip")




<p id="gdcalert20" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image20.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert21">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image20.png "image_tooltip")


<span style="text-decoration:underline;">Node and Edge counts</span> (GTEXCOEXP)



<p id="gdcalert21" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image21.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert22">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image21.png "image_tooltip")


For GTEXCOEXP, all nodes are existing HGNC. Edges and counts listed below.


<table>
  <tr>
   <td><strong>Predicate</strong>
   </td>
   <td><strong>Count</strong>
   </td>
   <td><strong>Predicate</strong>
   </td>
   <td><strong>Count</strong>
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Adipose___Subcutaneous</strong>
   </td>
   <td>15485
   </td>
   <td><strong>coexpression_Esophagus___Gastroesophageal_Junction</strong>
   </td>
   <td>12874
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Adipose___Visceral_(Omentum)</strong>
   </td>
   <td>2646
   </td>
   <td><strong>coexpression_Esophagus___Mucosa</strong>
   </td>
   <td>20463
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Adrenal_Gland</strong>
   </td>
   <td>37897
   </td>
   <td><strong>coexpression_Esophagus___Muscularis</strong>
   </td>
   <td>79416
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Artery___Aorta</strong>
   </td>
   <td>642521
   </td>
   <td><strong>coexpression_Fallopian_Tube</strong>
   </td>
   <td>769999
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Artery___Coronary</strong>
   </td>
   <td>612950
   </td>
   <td><strong>coexpression_Heart___Atrial_Appendage</strong>
   </td>
   <td>168057
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Artery___Tibial</strong>
   </td>
   <td>10237
   </td>
   <td><strong>coexpression_Heart___Left_Ventricle</strong>
   </td>
   <td>600676
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Bladder</strong>
   </td>
   <td>529181
   </td>
   <td><strong>coexpression_Kidney___Cortex</strong>
   </td>
   <td>583782
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Amygdala</strong>
   </td>
   <td>22221
   </td>
   <td><strong>coexpression_Kidney___Medulla</strong>
   </td>
   <td>10461695
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Anterior_cingulate_cortex_(BA24)</strong>
   </td>
   <td>102887
   </td>
   <td><strong>coexpression_Liver</strong>
   </td>
   <td>20645
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Caudate_(basal_ganglia)</strong>
   </td>
   <td>7309
   </td>
   <td><strong>coexpression_Lung</strong>
   </td>
   <td>17156
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Cerebellar_Hemisphere</strong>
   </td>
   <td>40983
   </td>
   <td><strong>coexpression_Minor_Salivary_Gland</strong>
   </td>
   <td>47164
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Cerebellum</strong>
   </td>
   <td>106195
   </td>
   <td><strong>coexpression_Muscle___Skeletal</strong>
   </td>
   <td>4061
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Cortex</strong>
   </td>
   <td>764276
   </td>
   <td><strong>coexpression_Nerve___Tibial</strong>
   </td>
   <td>12460
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Frontal_Cortex_(BA9)</strong>
   </td>
   <td>27760
   </td>
   <td><strong>coexpression_Ovary</strong>
   </td>
   <td>20177
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Hippocampus</strong>
   </td>
   <td>84051
   </td>
   <td><strong>coexpression_Pancreas</strong>
   </td>
   <td>24183
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Hypothalamus</strong>
   </td>
   <td>185487
   </td>
   <td><strong>coexpression_Pituitary</strong>
   </td>
   <td>28152
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Nucleus_accumbens_(basal_ganglia)</strong>
   </td>
   <td>1198329
   </td>
   <td><strong>coexpression_Prostate</strong>
   </td>
   <td>4197
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Putamen_(basal_ganglia)</strong>
   </td>
   <td>1146393
   </td>
   <td><strong>coexpression_Skin___Not_Sun_Exposed_(Suprapubic)</strong>
   </td>
   <td>1516
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Spinal_cord_(cervical_c_1)</strong>
   </td>
   <td>267533
   </td>
   <td><strong>coexpression_Skin___Sun_Exposed_(Lower_leg)</strong>
   </td>
   <td>84793
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Brain___Substantia_nigra</strong>
   </td>
   <td>143792
   </td>
   <td><strong>coexpression_Small_Intestine___Terminal_Ileum</strong>
   </td>
   <td>73157
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Breast___Mammary_Tissue</strong>
   </td>
   <td>3094
   </td>
   <td><strong>coexpression_Spleen</strong>
   </td>
   <td>375064
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Cells___Cultured_fibroblasts</strong>
   </td>
   <td>11652
   </td>
   <td><strong>coexpression_Stomach</strong>
   </td>
   <td>12964
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Cells___EBV_transformed_lymphocytes</strong>
   </td>
   <td>90051
   </td>
   <td><strong>coexpression_Testis</strong>
   </td>
   <td>141440
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Cervix___Ectocervix</strong>
   </td>
   <td>817624
   </td>
   <td><strong>coexpression_Thyroid</strong>
   </td>
   <td>31593
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Cervix___Endocervix</strong>
   </td>
   <td>805252
   </td>
   <td><strong>coexpression_Uterus</strong>
   </td>
   <td>16498
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Colon___Sigmoid</strong>
   </td>
   <td>11358
   </td>
   <td><strong>coexpression_Vagina</strong>
   </td>
   <td>20584
   </td>
  </tr>
  <tr>
   <td><strong>coexpression_Colon___Transverse</strong>
   </td>
   <td>22786
   </td>
   <td><strong>coexpression_Whole_Blood</strong>
   </td>
   <td>13845
   </td>
  </tr>
</table>



## The Human BioMolecular Atlas Program (HuBMAP)


### HuBMAP data sets


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>AZ, HUBMAP
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td><a href="https://hubmapconsortium.org/">https://hubmapconsortium.org/</a>
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>HuBMAP
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Jonathan Silverstein, Phil Blood
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>HuBMAP data provides tissue, cell-type and gene specific markers from single-cell data. The purpose of the Hubmap/AZ data is to provide cell-type-specific gene expression markers from single-cell experiments across each tissue.
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>HuBMAP is working to catalyze the development of a framework for  \
mapping the human body at single cell resolution and developing the tools to create an open, global atlas of the human body at the cellular level. In this database, we include cell-type specific gene markers from the Azimuth project form a subset of tissues including heart, liver and kidney.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td>See <a href="https://azimuth.hubmapconsortium.org/references/">https://azimuth.hubmapconsortium.org/references/</a>
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td>https://azimuth.hubmapconsortium.org/references/
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>769
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>910
   </td>
  </tr>
</table>



### HuBMAP Az Schema Diagram



<p id="gdcalert22" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image22.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert23">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image22.png "image_tooltip")



### HuBMAP Az Node and Edge counts


## 

<p id="gdcalert23" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image23.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert24">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image23.png "image_tooltip")


<p id="gdcalert24" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image24.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert25">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image24.png "image_tooltip")



## Illuminating the Druggable Genome (IDG)


### IDG Datasets


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>IDGP (compound/protein), IDGD (compound/disease) 
<p>
(both are edge SABs)
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td><a href="https://pharos.nih.gov/">https://pharos.nih.gov/</a> <a href="https://commonfund.nih.gov/idg">https://commonfund.nih.gov/idg</a> 
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>Illuminating the Druggable Genome Data Coordinating Center - Engagement Plan with the CFDE
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Christophe Lambert (PI), University of New Mexico Health Sciences Center
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>Relationships between compounds, diseases, and proteins drawn from the IDG Target Central Resource Database (TCRD) hosted at <a href="https://pharos.nih.gov">https://pharos.nih.gov</a> and at DrugCentral <a href="https://drugcentral.org/">https://drugcentral.org/</a>.
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>The Illuminating the Druggable Genome (IDG) project elucidates the relationships between diseases, targets, and compounds, providing insights into lesser-known proteins, empowering researchers to discover novel therapeutic targets and accelerate drug development for various diseases.
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>The IDG contributions to the Knowledge Graph include compounds, diseases and proteins and their relationships. A full description of IDG data sources is here: <a href="https://pharos.nih.gov/about">https://pharos.nih.gov/about</a> 
<p>
Target Central Resource Database (TCRD) is the central resource supporting the IDG-KMC.  TCRD has information about human drug targets with a focus on GPCRs, kinases, and ion channels.  TCRD categorizes all drug targets into four Target Development Levels (TDLs) by making use of activity thresholds.  Protein drug targets from TCRD with known bioactive compounds were incorporated into the IDG KG.  Also included in our KG are diseases from TCRD that have known “indication” relationships to approved drugs from DrugCentral.  The IDG KG can be used to explore compounds and proteins related to a specific disease among other similar queries.  The IDG KG can be combined with data from other DCCs such as LINCS, GTeX, and others to create interesting scientific use cases.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td>Compound nodes were sourced using TCRD, DrugCentral, and PubChem using PUBCHEM_CID ontology.  Specifically, chemical compounds from DrugCentral were included if TCRD indicated known bioactivity against protein targets.  Compound node properties include SMILES as the <em>node_definition</em>, drugbank ID as <em>node_dbxrefs</em>, name as <em>node_label</em>, and ‘IDG’ as the <em>node_namespace</em>.  The PubChem API was used to assign the <em>node_synonyms</em> and <em>node_dbxrefs</em> node properties.
<p>
Protein nodes were obtained from TCRD and DrugCentral, and rely on UNIPROTKB ontology.  Protein targets from DrugCentral and TCRD with known bioactive compounds were included.  A protein’s symbol is denoted as <em>node_label</em>, protein names as <em>node_definition</em>, EnsEMBL IDs as <em>node_dbxrefs</em>, and ‘IDG’ as <em>node_namespace</em>.
<p>
The disease nodes use SNOMED_US ontology and are sourced from TCRD and DrugCentral. Diseases from DrugCentral and TCRD with known indication relationships to approved drugs were included.  The disease OMOP concept names are included as the node property <em>node_label</em>.  Additionally, OMOP IDs are included as <em>node_dbxrefs</em> and ‘IDG’ as the <em>node_namespace</em>.
<p>
The bioactivity relationship is defined between compounds and proteins.  While ChEMBL and PubChem offer complex and differing ontologies for bioactivity relationships, for the sake of simplicity and efficiency in this early version we use the custom term "bioactivity". The type of bioactivity measurement (e.g. IC50, Kd, EC50) is included as <em>evidence_class</em>.
<p>
The indication relationship links compounds and diseases.  For simplicity, we introduce the custom, simple term “indication”.  These indication relationships are defined between diseases and approved drugs from DrugCentral. 
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td>Code by IDG team:
<p>
<a href="https://github.com/unmtransinfo/cfde-distillery">https://github.com/unmtransinfo/cfde-distillery</a> 
<p>
Code from core DD team (further processing and formatting):
<p>
<a href="https://github.com/TaylorResearchLab/CFDE_DataDistillery/blob/main/DCC_workflows/IDG/check_IDG.ipynb">https://github.com/TaylorResearchLab/CFDE_DataDistillery/blob/main/DCC_workflows/IDG/check_IDG.ipynb</a>
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>Total: 331788; compounds: 327951; disease: 1472; protein: 2365
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>Total: 463972; compound/protein (Bioactivity): 454957; compound/disease (Indication): 9015
   </td>
  </tr>
  <tr>
   <td>Source Data DOI(s) <em>(optional)</em>
   </td>
   <td>Raw data DOI
   </td>
  </tr>
  <tr>
   <td>Source Data URL(s) <em>(optional)</em>
   </td>
   <td><a href="https://app.globus.org/file-manager?origin_id=24c2ee95-146d-4513-a1b3-ac0bfdb7856f&origin_path=%2Fprojects%2Fdata-distillery%2FImport%2FIDG%2F">https://app.globus.org/file-manager?origin_id=24c2ee95-146d-4513-a1b3-ac0bfdb7856f&origin_path=%2Fprojects%2Fdata-distillery%2FImport%2FIDG%2F</a> 
   </td>
  </tr>
</table>



### IDG Schema Diagram



<p id="gdcalert25" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image25.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert26">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image25.png "image_tooltip")



### IDGP Node and Edge counts (compound/protein, **SAB = IDGP**)



<p id="gdcalert26" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image26.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert27">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image26.png "image_tooltip")


<p id="gdcalert27" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image27.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert28">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image27.png "image_tooltip")



### IDGD Node and Edge counts (compound/disease, **SAB = IDGD**)



<p id="gdcalert28" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image28.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert29">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image28.png "image_tooltip")




<p id="gdcalert29" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image29.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert30">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image29.png "image_tooltip")


For IDGP and IDGD, the numbers shown in the tables are numbers of IDG edges connecting existing SAB nodes as indicated.


## Gabriella Miller Kids First (GMKF)


### KF Datasets


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>KFGENEBIN, KFPT, KFCOHORT
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td><a href="https://kidsfirstdrc.org/">https://kidsfirstdrc.org/</a>
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>Gabriella Miller Kids First (GMKF) Pediatric Research Program Data Resource Center (DRC)
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Deanne Taylor (PI), Children’s Hospital of Philadelphia
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>Genomic and phenotypic data, broadly summarized from trio cohorts with cardiac birth defects from the Pediatric Cardiac Genetics Consortium cohort in Kids First,cohort SD_PREASA7S
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>The main purpose of the KF DRC is to better understand the genetic causes and links between childhood cancer and structural birth defects.
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>The Kids First DRC is a collaborative pediatric research effort created to accelerate data-driven discoveries and the development of novel precision-based approaches for children diagnosed with cancer or a structural birth defect using large genomic datasets. The Kids First DRC is comprised of integrated core teams that support development of leading-edge big data infrastructure and provide the necessary resources and tools to empower researchers and clinicians. 
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td>Variant data from a Congenital Heart Defects (CHD) cohort was queried and filtered using the Kids First variant workbench platform. We filtered for variants that were scored as ‘high impact’ by the variant effect predictor tool (VEP). Variant counts per gene were then computed by counting how many times each gene appeared. The number of variations per gene is stored in the ‘value’ property of the SAB <strong>KFGENEBIN</strong> Code nodes. 
<p>
Kids First cohorts are stored in the graph as their own nodes and have an SAB of <strong>KFCOHORT.</strong>
<p>
Patient IDs from the CHD cohort  have also been ingested into the graph as their own nodes and have an SAB of <strong>KFPT</strong>.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td><a href="https://github.com/TaylorResearchLab/CFDE_DataDistillery/tree/main/DCC_workflows/KidsFirst">https://github.com/TaylorResearchLab/CFDE_DataDistillery/tree/main/DCC_workflows/KidsFirst</a>
   </td>
  </tr>
  <tr>
   <td>Total nodes (rows)
   </td>
   <td>18,719
   </td>
  </tr>
  <tr>
   <td>Total edges (rows)
   </td>
   <td>76,690
   </td>
  </tr>
  <tr>
   <td>Source Data DOI(s) <em>(optional)</em>
   </td>
   <td>Raw data DOI
   </td>
  </tr>
  <tr>
   <td>Source Data URL(s) <em>(optional)</em>
   </td>
   <td>Raw data URL
   </td>
  </tr>
</table>



### KF Schema Diagram



<p id="gdcalert30" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image30.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert31">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image30.png "image_tooltip")



### KF Node and Edge counts



<p id="gdcalert31" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image31.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert32">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image31.png "image_tooltip")




<p id="gdcalert32" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image32.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert33">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image32.png "image_tooltip")



## The Library of Integrated Network-Based Cellular Signatures (LINCS)


### LINCS datasets


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>LINCS (edge SAB only)
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td><a href="https://lincsproject.org/">https://lincsproject.org/</a> 
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>Library of Integrated Network-Based Cellular Signatures (LINCS) Data Coordination and Integration Center (DCIC)
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Avi Ma’ayan (PI), Icahn School of Medicine at Mount Sinai
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>Gene expression changes resulting from drug/small molecule perturbations across cell lines, and gene expression signature similarity between drug/small molecule based on LINCS L1000 signature similarity
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>Understand cellular responses to various drug and pre-clinical compound treatments through L1000 transcriptomics assays
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>The LINCS assertions include drug-gene associations and drug-drug similarity associations computed from the LINCS L1000 consensus signatures dataset. Each drug is linked to the top 25 most up-regulated and top 25 most down-regulated genes in the L1000 consensus signatures for the drug/small-molecule, as well as to the top 5 most similar other drugs in the dataset based on the correlation between the consensus signatures for each drug. 
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td>Level 3 L1000 profiles, drug metadata, and gene metadata were first downloaded from CLUE.io. The L1000 Level 5 signatures were then computed using the Characteristic Direction method [BMC Bioinformatics 15, 79 (2014)]. For each signature, replicate L1000 profiles for a given perturbagen and dosage were compared against all other L1000 profiles from the same cell line batch. Consensus signatures for each drug were then computed by taking the mean of all gene expression vectors corresponding to the given drug across cell lines, timepoints, and dosages. 
<p>
Drugs were filtered to only those with known PubChem IDs in the original CLUE.io metadata, resulting in a final set of 4,523 drugs. The top 25 up- and down-regulated genes in each consensus signature with known Ensembl IDs from the metadata were determined by the greatest positive and negative Characteristic Direction coefficients, respectively. In total, 225,509 edges and 4,419 unique genes are represented in this collection of knowledge graph assertions. 
<p>
Additionally, a drug-drug similarity matrix was generated by computing the cosine similarity between all possible pairs of the consensus drug signatures. For each drug, the top 5 other drugs with the greatest positive cosine similarity values were retained. Duplicate edges were removed, resulting in 20,785 total edges representing consensus signature-based drug-drug similarity between the 4,523 drugs with known PubChem IDs. 
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td>The methods are described in the following publication:
<p>
Evangelista, J.E., Clarke, D.J.B., Xie, Z. et al. Toxicology knowledge graph for structural birth defects. Commun Med 3, 98 (2023).
<p>
The code to produce the assertions can be found at: 
<p>
<a href="https://github.com/nih-cfde/ReproToxTables">https://github.com/nih-cfde/ReproToxTables</a> 
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>8,942 (drugs: 4523; genes: 4419)
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>246,294 (drug-gene: 225,509; drug-drug: 20,785)
   </td>
  </tr>
  <tr>
   <td>Source Data DOI(s) <em>(optional)</em>
   </td>
   <td>N/A
   </td>
  </tr>
  <tr>
   <td>Source Data URL(s) <em>(optional)</em>
   </td>
   <td><a href="https://maayanlab.cloud/sigcom-lincs/#/Download">https://maayanlab.cloud/sigcom-lincs/#/Download</a> 
   </td>
  </tr>
</table>



### LINCS Schema Diagram<span style="text-decoration:underline;">  \
</span>

<p id="gdcalert33" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image33.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert34">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image33.png "image_tooltip")



### Node and Edge counts



<p id="gdcalert34" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image34.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert35">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image34.png "image_tooltip")




<p id="gdcalert35" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image35.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert36">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image35.png "image_tooltip")


For the SAB LINCS, the numbers shown in the tables are numbers of edges connecting existing SAB nodes as indicated.


## The Molecular Transducers of Physical Activity Consortium (MoTrPAC)


### MoTrPAC datasets


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>MOTRPAC
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td>https://motrpac-data.org
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>Molecular Transducers of Physical Activity Consortium (MoTrPAC) Bioinformatics Center (BIC)
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Euan Ashley MD PhD (PI), Matthew Wheeler MD PhD (PI)
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>Gene differential expression changes resulting from the RNA-seq data of young adult rats (6 month old) performing endurance training exercise at the 1 week, 2 week, 4 week and 8 week time points.
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>The Molecular Transducers of Physical Activity Consortium (MoTrPAC) aims to elucidate how exercise improves health and ameliorates diseases by building a map of the molecular responses to endurance exercise.
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>MoTrPAC is a multi-site collaboration across the US encompassing various scientific disciplines: preclinical animal study sites and human clinical exercise sites, which perform the exercise testing and biospecimen collection; a consortium coordinating center and biorepository, which manages sample collection, distribution of samples, and consortium logistics; chemical analysis sites, which are responsible for omics analysis from the samples collected; and a bioinformatics center to collaboratively analyze and map the data generated by the other sites along with data dissemination to make the data and other resources available to the public. The animal studies enable analysis of the effects of exercise on many different tissues that are not readily obtainable in humans, whereas the collection of accessible human tissues (muscle, blood, and adipose) will permit the analysis of the direct effect of exercise in humans. Additional information can be found at the main consortium page (https://motrpac.org) or at the data portal (<a href="https://motrpac-data.org">https://motrpac-data.org</a>).
<p>
The MoTrPAC study is divided into two main parts - animal (rats) and human, with multiple phases or interventions in each of them.   Preclinical animal study sites conduct the endurance exercise and training intervention in young adult (6 month old) and middle-aged adult (18 month old) rats, while Clinical study sites conduct the human endurance and resistance training interventions in pediatric, adults and highly active adults.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td>
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td><a href="https://github.com/TaylorResearchLab/CFDE_DataDistillery/blob/main/DCC_workflows/MoTrPAC/MOTRPAC.ipynb">https://github.com/TaylorResearchLab/CFDE_DataDistillery/blob/main/DCC_workflows/MoTrPAC/MOTRPAC.ipynb</a>
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>16149
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>25714
   </td>
  </tr>
  <tr>
   <td>Source Data DOI(s) <em>(optional)</em>
   </td>
   <td>Raw data DOI
   </td>
  </tr>
  <tr>
   <td>Source Data URL(s) <em>(optional)</em>
   </td>
   <td>Raw data URL
   </td>
  </tr>
</table>



### MOTRPAC Schema Diagram

<p id="gdcalert36" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image36.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert37">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image36.png "image_tooltip")


<span style="text-decoration:underline;">Node and Edge counts</span>



<p id="gdcalert37" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image37.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert38">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image37.png "image_tooltip")


<p id="gdcalert38" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image38.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert39">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image38.png "image_tooltip")



## Metabolomics Workbench (MW)


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>MW (REFMET; metabolite nodes)
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td>https://www.metabolomicsworkbench.org/
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>Metabolomics Workbench
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Professor Shankar Subramaniam (PI)
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>Gene-metabolite relationships: MW database tables based on KEGG and other resources
<p>
Disease-metabolite relationships: Publication based on HMDB data (<a href="https://pubmed.ncbi.nlm.nih.gov/32426349/">https://pubmed.ncbi.nlm.nih.gov/32426349/</a>)
<p>
Cell-metabolite relationships: MW database tables for data submitted to NMDR
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>Understand what metabolites may be regulated by various genes and their spatial (anatomical) and disease context.
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>The<a href="https://commonfund.nih.gov/metabolomics/index"> National Institutes of Health (NIH) Common Fund Metabolomics Program</a> was developed with the goal of increasing national capacity in metabolomics by supporting the development of next generation technologies, providing training and mentoring opportunities, increasing the inventory and availability of high quality reference standards, and promoting data sharing and collaboration. In support of this effort, the Metabolomics Common Fund's National Metabolomics Data Repository(NMDR), housed at the San Diego Supercomputer Center (SDSC), University of California, San Diego, has developed the Metabolomics Workbench. The Metabolomics Workbench serves as a national and international repository for metabolomics data and metadata and provides analysis tools and access to metabolite standards, protocols, tutorials, training, and more.
<p>
NMDR houses data on metabolomics studies conducted by various centers and research laboratories across the nation and the world, spanning many species, sample sources, diseases, metabolomics experimental techniques and metabolite classes [https://www.metabolomicsworkbench.org/data/browse.php].  The data we have shared with the Data Distillery partnership is a key subset of all the data in NMDR, centered around metabolites. Specifically, we have shared disease-metabolite, gene-metabolite and cell/anatomy (sample source)-metabolite relationships, which when integrated with data from other DCC and external resources has the potential to address interesting biological questions.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td><span style="text-decoration:underline;">Gene-Metabolite: </span>Human genes catalyzing metabolic reactions and their associated metabolites were obtained from MW database tables. The HGNC ID was used as metabolic gene <em>node_id</em>, and its approved symbol and name are used as <em>node_label</em> and <em>node_definition</em>, respectively. UMLS, ENTREZ and ENSMBL IDs are used as <em>node_dbxrefs</em>. For the edges, the Subject (Gene: HGNC ID) was related to the Object (Metabolite: PUBCHEM_CID) by the Predicate (RO_0002566: Causally influences).
<p>
<span style="text-decoration:underline;">Disease-Metabolite:</span> Disease-metabolite entities and relationships were deduced from the publication (PMID: 32426349) based on HMDB. The PUBCHEM_CID/HMDB ID was used as <em>node_id</em> for the metabolite. Similarly, disease entities were encoded with DOID or HPO IDs. UMLS, PUBCHEM_CID, DRUGBANK and REFMET were used as <em>node_dbxrefs. </em>For the edges,<em> </em>the Subject (Metabolite: PUBCHEM_CID/HMDB ID) was related to the Object (Disease: DOID/HPO) by the Predicate (RO_0003308: Correlated with condition).
<p>
<span style="text-decoration:underline;">Cell-Metabolite: </span>Metabolite-anatomy context (cell/tissue association) was obtained from MW database. Cell/tissue entity <em>node_id </em>is encoded with UBERON, CL and CLO IDs and cross referenced with UMLS. For the edges, the Subject (Spatial context: UBERON/CL/CLO) was related to the Object (Metabolite: PUBCHEM_CID) by the Predicate (RO_0003000: Produces).
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td><a href="https://github.com/mano-at-sdsc/MW_DataDistillery">https://github.com/mano-at-sdsc/MW_DataDistillery</a> 
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>51271
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>10009
   </td>
  </tr>
  <tr>
   <td>Source Data DOI(s) <em>(optional)</em>
   </td>
   <td>Raw data DOI
   </td>
  </tr>
  <tr>
   <td>Source Data URL(s) <em>(optional)</em>
   </td>
   <td>Raw data URL
   </td>
  </tr>
</table>


<span style="text-decoration:underline;">MW Schema Diagram \
</span>

<p id="gdcalert39" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image39.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert40">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image39.png "image_tooltip")


<span style="text-decoration:underline;">Node and Edge Counts \
</span>


<table>
  <tr>
   <td><strong>SAB</strong>
   </td>
   <td><strong>Count</strong>
   </td>
  </tr>
  <tr>
   <td><strong>UBERON</strong>
   </td>
   <td>67
   </td>
  </tr>
  <tr>
   <td><strong>CL</strong>
   </td>
   <td>12
   </td>
  </tr>
  <tr>
   <td><strong>CLO</strong>
   </td>
   <td>3
   </td>
  </tr>
  <tr>
   <td><strong>HGNC</strong>
   </td>
   <td>1061
   </td>
  </tr>
  <tr>
   <td><strong>PUBCHEM</strong>
   </td>
   <td>8543
   </td>
  </tr>
  <tr>
   <td><strong>HMDB</strong>
   </td>
   <td>18
   </td>
  </tr>
  <tr>
   <td><strong>DOID</strong>
   </td>
   <td>276
   </td>
  </tr>
  <tr>
   <td><strong>HPO</strong>
   </td>
   <td>32
   </td>
  </tr>
</table>



<table>
  <tr>
   <td><strong>Subject SAB</strong>
   </td>
   <td><strong>Predicate</strong>
   </td>
   <td><strong>Object SAB</strong>
   </td>
   <td><strong>Count</strong>
   </td>
  </tr>
  <tr>
   <td><strong>UBERON</strong>
   </td>
   <td>produces
   </td>
   <td>PUBCHEM
   </td>
   <td>34471
   </td>
  </tr>
  <tr>
   <td><strong>CL</strong>
   </td>
   <td>produces
   </td>
   <td>PUBCHEM
   </td>
   <td>6777
   </td>
  </tr>
  <tr>
   <td><strong>CLO</strong>
   </td>
   <td>produces
   </td>
   <td>PUBCHEM
   </td>
   <td>536
   </td>
  </tr>
  <tr>
   <td><strong>HGNC</strong>
   </td>
   <td>causally influences
   </td>
   <td>PUBCHEM
   </td>
   <td>5527
   </td>
  </tr>
  <tr>
   <td><strong>PUBCHEM</strong>
   </td>
   <td>correlated with condition
   </td>
   <td>DOID
   </td>
   <td>3856
   </td>
  </tr>
  <tr>
   <td><strong>HMDB</strong>
   </td>
   <td>correlated with condition
   </td>
   <td>DOID
   </td>
   <td>27
   </td>
  </tr>
  <tr>
   <td><strong>PUBCHEM</strong>
   </td>
   <td>correlated with condition
   </td>
   <td>HPO
   </td>
   <td>77
   </td>
  </tr>
</table>


For the SAB MW, the numbers shown in the tables are numbers of edges connecting existing SAB nodes as indicated.


## Stimulating Peripheral Activity to Relieve Conditions (SPARC)


<table>
  <tr>
   <td>Dataset SAB(s)
   </td>
   <td>SCKAN, NPO, UBERON, PATO, NIFSTD
   </td>
  </tr>
  <tr>
   <td>DCC Website
   </td>
   <td>https://sparc.science/
   </td>
  </tr>
  <tr>
   <td>DCC
   </td>
   <td>SPARC Data and Resouce Center (DRC) - Knowledge Management and Curation Core
   </td>
  </tr>
  <tr>
   <td>Authority
   </td>
   <td>Tom Gillespie, Fahim Imam (SPARC K-Core, University of California San Diego)
<p>
Jyl Boline (PM - SPARC K-Core)
   </td>
  </tr>
  <tr>
   <td>Source Information
   </td>
   <td>A key component of the SPARC Program is the SPARC Connectivity Knowledge Base of the Autonomic Nervous system, referred to as SCKAN. SCKAN is a semantic store housing a comprehensive knowledge base of autonomic nervous system (ANS) nerve to end organ connectivity. Connectivity information is derived from SPARC experts, SPARC data, and the literature and textbooks using a Natural Language Processing (NLP) pipeline. 
   </td>
  </tr>
  <tr>
   <td>Purpose
   </td>
   <td>Facilitate enhanced understanding of the peripheral nervous system to support the development of effective bioelectronic therapies by driving collaborative neurosciences and providing online resources for accessing and submitting curated data and models, as well as dynamic knowledge-management and visualization tools.
   </td>
  </tr>
  <tr>
   <td>Description
   </td>
   <td>The SPARC Knowledge base of the Automatic Nervous System (SCKAN) is an integrated graph database composed of three parts: the SPARC dataset metadata graph, <a href="http://open-physiology-viewer-docs.surge.sh/">ApiNATOMY</a> and Neuron Phenotype Ontology (<a href="https://www.biorxiv.org/content/10.1101/2020.09.01.278879v1.full">NPO</a>) models of connectivity, and the larger ontology used  by SPARC which is a combination of the NIF-Ontology and community ontologies.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology
   </td>
   <td>SCKAN provides a central location to populate, discover, and query ANS connectivity knowledge over multiple scales.  It allows issuing queries such as, “what are the locations of neuron somas with processes that pass through spinal cord level C4?” and create a searchable visual atlas of ANS circuitry. Users of the <a href="https://sparc.science/maps">SPARC maps</a> can query SCKAN to find more information about routes, targets and evidence.
<p>
SCKAN contains statements about neuronal connectivity at the neuron population level, largely in the form of: “Neurons with somas in structure A project to structure B via nerve C.” SCKAN models connections at two levels of granularity: circuits and individual connections. A circuit represents a detailed model of connectivity that is associated with a particular organ like  bladder or functional circuits like defensive breathing. Circuits contain detailed representations of neuron populations giving rise to ANS connections. They include mappings of the locations of cell bodies, dendrites, axon segments as well as synaptic endings involved in a particular circuit. 
<p>
Circuits in SCKAN are modelled using  ApiNATOMY, a knowledge model and a tool specifically created to represent multiscale connectivity. To provide a comprehensive knowledge about ANS connectivity, the circuit-based approach is supplemented with well-known connections of ANS derived from the literature and textbooks using a Natural Language Processing (NLP) pipeline.These types of individual connectivity statements do not have detailed topological information associated with them and are represented using NPO.
   </td>
  </tr>
  <tr>
   <td>Summarization Methodology Code Repository URL
   </td>
   <td>https://zenodo.org/record/7476115
   </td>
  </tr>
  <tr>
   <td>Total nodes
   </td>
   <td>484,768
   </td>
  </tr>
  <tr>
   <td>Total edges
   </td>
   <td>1,337,124
   </td>
  </tr>
  <tr>
   <td>Source Data DOI(s) <em>(optional)</em>
   </td>
   <td>10.5281/zenodo.7476115
   </td>
  </tr>
  <tr>
   <td>Source Data URL(s) <em>(optional)</em>
   </td>
   <td>
<ul>

<li><a href="https://doi.org/10.5281/zenodo.7476115">https://doi.org/10.5281/zenodo.7476115</a>

<li><a href="https://github.com/open-physiology/apinatomy-models">https://github.com/open-physiology/apinatomy-models</a> 

<li><a href="https://github.com/SciCrunch/NIF-Ontology/">https://github.com/SciCrunch/NIF-Ontology/</a>

<li><a href="https://bioportal.bioontology.org/ontologies/NPOKB">https://bioportal.bioontology.org/ontologies/NPOKB</a> 
</li>
</ul>
   </td>
  </tr>
</table>


<span style="text-decoration:underline;">SPARC Schema Diagram</span>


## 

<p id="gdcalert40" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image40.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert41">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image40.png "image_tooltip")


<span style="text-decoration:underline;">Node and Edge counts</span>


## 

<p id="gdcalert41" ><span style="color: red; font-weight: bold">>>>>>  gd2md-html alert: inline image link here (to images/image41.png). Store image on your image server and adjust path/filename/extension if necessary. </span><br>(<a href="#">Back to top</a>)(<a href="#gdcalert42">Next alert</a>)<br><span style="color: red; font-weight: bold">>>>>> </span></p>


![alt_text](images/image41.png "image_tooltip")



# Additional Datasets


## CLINVAR

The ClinVar dataset (v2023-01-05) was utilized to define assertions between human genes and phenotypes. Only genes with pathogenic, likely pathogenic and pathogenic/likely pathogenic variants were considered, and we excluded associations with no assertion criteria met. To retrieve the target phenotype/disease we used MedGen IDs listed in the ClinVar dataset (also already present in the KG). Processed ClinVar dataset contains 214,040 relationships (including reverse relationships) with the following characteristics [Type: “gene_associated_with_disease_or_phenotype”, SAB: “CLINVAR”] and [type: inverse_gene_associated_with_disease_or_phenotype, S AB: “CLINVAR”] connecting HGNC to MONDO, HPO, EFO and MESH Concept nodes. 


## CMAP

The edge lists of the CMAP Signatures of Differentially Expressed Genes for Small Molecules dataset were obtained from the Harmonizome database [https://maayanlab.cloud](https://maayanlab.cloud). The dataset added 2,625,336 relationships (including reverse relationships) connecting the CHEBI and HGNC nodes with predicates “negatively_correlated_with_gene”, “inverse_negatively_correlated_with_gene”, “positively_correlated_with_gene”, “inverse_positively_correlated_with_gene” (SAB: “CMAP”).


## HPOMP

This set of assertions maps human phenotype ontology (HPO) nodes to mammalian phenotype ontology (MP) nodes through the ‘is_approximately_equivalent_to’. It is essentially a set of assertions mapping human phenotype codes to mouse phenotype codes. The mappings were produced by using a software tool called [PheKnowLator](https://github.com/callahantiff/PheKnowLator). There are 1,785 HPOMP mappings. These assertions can be queried by specifying the SAB property as HPOMP on the ‘is_approximately_equivalent_to’ relationship.


## HGNCHPO

This set of assertions maps HGNC gene nodes to human phenotype ontology (HPO) nodes through the ‘associated_with’ relationship. There are 671,046 HGNCHPO mappings. These assertions can be queried by specifying the SAB property as HGNCHPO on the ‘associated_with’ relationship.


## HCOPHGNC

This set of assertions maps mouse gene nodes (HCOP) to human gene nodes (HGNC). Mouse gene nodes are referred to as ‘HCOP’ in the Data Distillery Knowledge Graph because the HGNC Comparison of Orthology Predictions ([HCOP](https://www.genenames.org/tools/hcop/)) tool was used to generate these mappings. The ‘in_1_to_1_orthology_relationship_with’ is used to connect the HGNC and HCOP nodes. There are 67,027 HCOPHGNC mappings. These assertions can be queried by specifying the SAB property as HCOPHGNC on the ‘in_1_to_1_orthology_relationship_with’ relationship.


## HCOPMP

This set of assertions maps mouse gene nodes (HCOP) to the mammalian phenotype ontology (MP) nodes through the ‘involved_in’ relationship. These mappings are the mouse version of the HGNCHPO mappings. Files from the International Mouse Phenotyping Consortium ([IMPC](https://www.mousephenotype.org/)) and Mouse Genome Informatics ([MGI](https://www.informatics.jax.org/)) were used to create this dataset. There are 234,043 HCOPMP mappings. These assertions can be queried by specifying the SAB property as HCOPMP on the ‘involved_in’ relationship.


## Homo Sapiens Chromosomal Location Ontology (HSCLO)

Homo Sapiens Chromosomal Location Ontology (HSCLO) was primarily created to connect 4DN loop coordinates to the rest of the graph through the mapping between HSCLO and GENCODE. HSCLO was later utilized to connect GTEXEQTL locations in the graph as searchable nodes at 1 kbp resolution (same as 4DN). The dataset relationships as well as nodes use HSCLO as their SAB. HSCLO nodes are defined at 5 resolution levels; chromosomes, 1 Mbp, 100 kbp, 10 kbp and 1kbp with each level connects to lower level with above_(resolution level)_band (e.g. “above_1Mbp_band”, “above 1_kbp_band”) and nodes at the same resolution level are connected through prcedes_(resolution level)_band (e.g. “precedes_10kbp_band”). The dataset contains 3,431,155 nodes and 6,862,195 relationships (13,724,390 bidirectional).


## MSIGDB

Five subsets of MSigDB v7.4 datasets were introduced as entity-gene relationships to the knowledge graph: C1 (positional gene sets), C2 (curated gene sets), C3 (regulatory target gene sets), C8 (cell type signature gene sets) and H (hallmark gene sets ). With this subset, MSIGDB Concept nodes were created for MSigDB systematic names (used as Codes excluding KEGG data). The relationships between these Concept nodes and HGNC nodes were defined using the mentioned 5 subsets where the subset information was included in the relationship SABs as “MSIGDB”. 


## RATHCOP

This set of assertions maps human ENSEMBL gene nodes to rat ENSEMBL gene nodes. These mappings were generated from the [HCOP](https://www.genenames.org/tools/hcop/) tool just like for the mouse to human assertions, except we used the ENSEMBL codes here instead of the HGNC codes. The ‘has_human_ortholog’ relationship is used to connect ENSEMBL Rat nodes to ENSEMBL Human nodes. There are 42,371 RATHCOP mappings and they can be queried by specifying the SAB property as RATHCOP on the ‘has_human_ortholog’ relationship.
