{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Building the UI\n",
    "In addition to the Neo4j console and the notebook earlier, we can provide develop a public facing website that we can share with the public without providing them write access to the neo4j database.\n",
    "\n",
    "The UI provides single and two-term search where the latter uses allShortestPath to query all the shortest paths between two terms. To build the UI, we need to create a UI schema that is built using the data that was ingested in the graph db. This notebook automates the creation of this schema file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import json\n",
    "from glob import glob\n",
    "import pandas as pd"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Node Schema\n",
    "\n",
    "```\n",
    "{\n",
    "    \"node\": \"Gene\",\n",
    "    \"example\": [\n",
    "        \"COPB2\",\n",
    "        \"SDF2L1\"\n",
    "    ],\n",
    "    \"display\": [\n",
    "        {\n",
    "            \"label\": \"Gene ID\",\n",
    "            \"text\": \"${id}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"label\",\n",
    "            \"text\": \"${label}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"uri\",\n",
    "            \"text\": \"${uri}\",\n",
    "            \"href\": \"${uri}\",\n",
    "            \"type\": \"link\"\n",
    "        }\n",
    "    ],\n",
    "    \"search\": [\n",
    "        \"id\",\n",
    "        \"label\"\n",
    "    ],\n",
    "    \"color\": \"#c5e1a5\"\n",
    "}\n",
    "\n",
    "```\n",
    "This is a node schema for the node \"Gene\". The example field will display as an example on the UI while the display field tells the UI what to display as a tooltip. Meanwhile, search tells the UI that we can search both id and label fields of the nodes. Node colors are automatically generated but can otherwise be set using the field color for more customization."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "node_schema = []\n",
    "node_pattern = \"(?P<directory>.+)/(?P<label>.+)\\.(?P<entity>.+)\\.csv\"\n",
    "for filename in glob(\"../out/filtered/nodes/*.nodes.csv\"):\n",
    "    match = re.match(node_pattern, filename).groupdict()\n",
    "    label = match[\"label\"].replace(\"_\", \" \")\n",
    "    df = pd.read_csv(filename, low_memory=False)\n",
    "    display = []\n",
    "    search = []\n",
    "    for i in df.columns:\n",
    "        if i == \"id\":\n",
    "            display.append({\n",
    "                'label': '%s ID'%label,\n",
    "                'text': \"${%s}\"%i,\n",
    "                'type': 'text'\n",
    "            })\n",
    "        else:\n",
    "            val = str(df.at[0, i])\n",
    "            if val.startswith(\"http\"):\n",
    "                display.append({\n",
    "                    'label': i,\n",
    "                    'text': \"${%s}\"%i,\n",
    "                    'href': \"${%s}\"%i,\n",
    "                    'type': 'link'\n",
    "                })\n",
    "            else:\n",
    "                display.append({\n",
    "                    'label': i,\n",
    "                    'text': \"${%s}\"%i,\n",
    "                    'type': 'text'\n",
    "                })\n",
    "    examples = list(df.head(2).label)\n",
    "    node = {\n",
    "        \"node\": label,\n",
    "        \"example\": examples,\n",
    "        \"display\": display,\n",
    "        \"relation\": [],\n",
    "        \"search\": [i for i in df.columns if \"value\" not in i and \"lowerbound\" not in i and \"upperbound\" not in i ],\n",
    "        \"color\": \"#c5e1a5\"\n",
    "    }\n",
    "    node_schema.append(node)\n",
    "        "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Edge Schema\n",
    "```\n",
    "{\n",
    "    \"match\": [\n",
    "        \"KEGG\"\n",
    "    ],\n",
    "    \"selected\": true,\n",
    "    \"display\": [\n",
    "        {\n",
    "            \"label\": \"source\",\n",
    "            \"text\": \"${source}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"relation\",\n",
    "            \"text\": \"${relation}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"target\",\n",
    "            \"text\": \"${target}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"source label\",\n",
    "            \"text\": \"${source_label}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"target label\",\n",
    "            \"text\": \"${target_label}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"resource\",\n",
    "            \"text\": \"${resource}\",\n",
    "            \"href\": \"${resource}\",\n",
    "            \"type\": \"link\"\n",
    "        }\n",
    "    ]\n",
    "}\n",
    "```\n",
    "\n",
    "This is the same with node schema with a few differences, first match allows you to match multiple edge types and selected is a field that allows us to select what edges are queried by default when doing two term searches."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edge_schema = []\n",
    "edge_pattern = \"(?P<directory>.+)/(?P<source_type>.+)\\.(?P<relation>.+)\\.(?P<target_type>.+)\\.(?P<entity>.+)\\.csv\"\n",
    "edge_dict = {}\n",
    "for filename in glob(\"../out/filtered/edges/*.edges.csv\"):\n",
    "   match = re.match(edge_pattern, filename).groupdict()\n",
    "   entity = match[\"entity\"]\n",
    "   source_type = match[\"source_type\"]\n",
    "   relation = match[\"relation\"].replace(\"_\", \" \")\n",
    "   edge = {\n",
    "      # \"match\": [relation],\n",
    "      \"selected\": True,\n",
    "   }\n",
    "   order = [\"evidence\", \"ASC\"] if \"negative\" in relation.lower() else [\"evidence\", \"DESC\"]\n",
    "   color = \"#bdbdbd\"\n",
    "   if \"positive\" in relation.lower():\n",
    "       color = \"#90caf9\" \n",
    "   elif \"negative\" in relation.lower():\n",
    "      color = \"#ef9a9a\"\n",
    "   edge[\"order\"] = order\n",
    "   edge[\"color\"] = color\n",
    "   df = pd.read_csv(filename)\n",
    "   display = []\n",
    "   for i in df.columns:\n",
    "      if \"label\" not in i:\n",
    "         if i == \"id\":\n",
    "            display.append({\n",
    "                  'label': '%s ID'%label,\n",
    "                  'text': \"${%s}\"%i,\n",
    "                  'type': 'text'\n",
    "            })\n",
    "         else:\n",
    "            val = str(df.at[0, i])\n",
    "            if val.startswith(\"http\"):\n",
    "                  display.append({\n",
    "                     'label': i.replace(\"_\", \" \"),\n",
    "                     'text': \"${%s}\"%i,\n",
    "                     'href': \"${%s}\"%i,\n",
    "                     'type': 'link'\n",
    "                  })\n",
    "            else:\n",
    "                  display.append({\n",
    "                     'label': i.replace(\"_\", \" \"),\n",
    "                     'text': \"${%s}\"%i,\n",
    "                     'type': 'text'\n",
    "                  })\n",
    "            edge[\"display\"] = display\n",
    "   edge_cols = \",\".join(list(df.columns) + order + [color])\n",
    "   if edge_cols not in edge_dict:\n",
    "       edge_dict[edge_cols] = {\n",
    "           \"match\": [relation],\n",
    "           **edge\n",
    "       }\n",
    "   else:\n",
    "       edge_dict[edge_cols][\"match\"].append(relation)\n",
    "\n",
    "edge_schema = list(edge_dict.values())\n",
    "len(edge_schema)\n",
    "         "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'match': ['coexpression Thyroid',\n",
       "  'expressed in',\n",
       "  'indication',\n",
       "  'coexpression Whole Blood',\n",
       "  'coexpression Skin   Not Sun Exposed (Suprapubic)',\n",
       "  'correlated with condition',\n",
       "  'coexpression Testis',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'indication',\n",
       "  'coexpression Heart   Atrial Appendage',\n",
       "  'coexpression Brain   Spinal cord (cervical c 1)',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'contains chromosome',\n",
       "  'correlated with condition',\n",
       "  'coexpression Brain   Putamen (basal ganglia)',\n",
       "  'coexpression Brain   Hypothalamus',\n",
       "  'correlated with condition',\n",
       "  'indication',\n",
       "  'coexpression Spleen',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'coexpression Kidney   Cortex',\n",
       "  'coexpression Brain   Cerebellum',\n",
       "  'coexpression Colon   Sigmoid',\n",
       "  'isa',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'is from source',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Brain   Nucleus accumbens (basal ganglia)',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'coexpression Adrenal Gland',\n",
       "  'coexpression Brain   Anterior cingulate cortex (BA24)',\n",
       "  'coexpression Artery   Tibial',\n",
       "  'correlated with condition',\n",
       "  'coexpression Minor Salivary Gland',\n",
       "  'coexpression Kidney   Medulla',\n",
       "  'coexpression Kidney   Cortex',\n",
       "  'precedes 1Mbp band',\n",
       "  'coexpression Brain   Hippocampus',\n",
       "  'correlated with condition',\n",
       "  'location of',\n",
       "  'has motif',\n",
       "  'coexpression Colon   Transverse',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'isa',\n",
       "  'correlated with condition',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'coexpression Fallopian Tube',\n",
       "  'correlated with condition',\n",
       "  'located in',\n",
       "  'coexpression Nerve   Tibial',\n",
       "  'correlated with condition',\n",
       "  'indication',\n",
       "  'expresses',\n",
       "  'coexpression Brain   Caudate (basal ganglia)',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'above 100kbp band',\n",
       "  'indication',\n",
       "  'has canonical residue',\n",
       "  'indication',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'refers to',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Skin   Sun Exposed (Lower leg)',\n",
       "  'correlated with condition',\n",
       "  'precedes 100kbp band',\n",
       "  'indication',\n",
       "  '4DN loop has qvalue bin',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Minor Salivary Gland',\n",
       "  'in similarity relationship with',\n",
       "  'correlated with condition',\n",
       "  'coexpression Adipose   Visceral (Omentum)',\n",
       "  'correlated with condition',\n",
       "  'coexpression Ovary',\n",
       "  'coexpression Kidney   Medulla',\n",
       "  'correlated with condition',\n",
       "  'coexpression Brain   Putamen (basal ganglia)',\n",
       "  'correlated with condition',\n",
       "  'location of',\n",
       "  'coexpression Cervix   Endocervix',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Artery   Aorta',\n",
       "  '4DN file has loop',\n",
       "  'correlated with condition',\n",
       "  'coexpression Lung',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'coexpression Uterus',\n",
       "  'coexpression Cells   Cultured fibroblasts',\n",
       "  'isa',\n",
       "  'coexpression Brain   Spinal cord (cervical c 1)',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'coexpression Brain   Putamen (basal ganglia)',\n",
       "  'coexpression Esophagus   Mucosa',\n",
       "  'attached by',\n",
       "  'correlated with condition',\n",
       "  'isa',\n",
       "  'coexpression Small Intestine   Terminal Ileum',\n",
       "  'correlated with condition',\n",
       "  'precedes 1kbp band',\n",
       "  'coexpression Brain   Cortex',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Vagina',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Bladder',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'coexpression Brain   Nucleus accumbens (basal ganglia)',\n",
       "  'correlated with condition',\n",
       "  'has expression',\n",
       "  'precedes 10kbp band',\n",
       "  'correlated with condition',\n",
       "  'coexpression Fallopian Tube',\n",
       "  'coexpression Bladder',\n",
       "  'coexpression Brain   Hypothalamus',\n",
       "  'isa',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'coexpression Cervix   Endocervix',\n",
       "  'coexpression Small Intestine   Terminal Ileum',\n",
       "  'coexpression Cervix   Endocervix',\n",
       "  '4DN loop us end',\n",
       "  'gene associated with disease or phenotype',\n",
       "  '4DN loop ds end',\n",
       "  'correlated with condition',\n",
       "  'coexpression Brain   Cerebellar Hemisphere',\n",
       "  'coexpression Brain   Hypothalamus',\n",
       "  'coexpression Brain   Substantia nigra',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'coexpression Muscle   Skeletal',\n",
       "  'isa',\n",
       "  'coexpression Bladder',\n",
       "  'isa',\n",
       "  'location of',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'coexpression Fallopian Tube',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Cells   EBV transformed lymphocytes',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'expressed in',\n",
       "  'p value',\n",
       "  'coexpression Adipose   Visceral (Omentum)',\n",
       "  'correlated with condition',\n",
       "  'expresses',\n",
       "  'coexpression Cells   EBV transformed lymphocytes',\n",
       "  'coexpression Cervix   Ectocervix',\n",
       "  'above 1kbp band',\n",
       "  'coexpression Brain   Amygdala',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'above 10kbp band',\n",
       "  'correlated with condition',\n",
       "  'coexpression Artery   Coronary',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  '4DN dataset has file',\n",
       "  'coexpression Adipose   Subcutaneous',\n",
       "  'coexpression Brain   Anterior cingulate cortex (BA24)',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'coexpression Heart   Left Ventricle',\n",
       "  'correlated with condition',\n",
       "  'indication',\n",
       "  'synthesized by',\n",
       "  'has glycosequence',\n",
       "  'coexpression Small Intestine   Terminal Ileum',\n",
       "  'indication',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'coexpression Cells   EBV transformed lymphocytes',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'coexpression Cervix   Ectocervix',\n",
       "  'coexpression Spleen',\n",
       "  'coexpression Brain   Cortex',\n",
       "  'correlated with condition',\n",
       "  'indication',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'indication',\n",
       "  'above 1Mbp band',\n",
       "  'has enzyme protein',\n",
       "  'coexpression Esophagus   Muscularis',\n",
       "  'location of',\n",
       "  'location of',\n",
       "  'coexpression Adipose   Visceral (Omentum)',\n",
       "  'correlated with condition',\n",
       "  'coexpression Brain   Cortex',\n",
       "  'isa',\n",
       "  'correlated with condition',\n",
       "  'coexpression Brain   Anterior cingulate cortex (BA24)',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Breast   Mammary Tissue',\n",
       "  'indication',\n",
       "  'indication',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Colon   Transverse',\n",
       "  'correlated with condition',\n",
       "  'coexpression Liver',\n",
       "  'coexpression Nerve   Tibial',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'has monosaccharide',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'indication',\n",
       "  'coexpression Pancreas',\n",
       "  'bioactivity',\n",
       "  'coexpression Brain   Spinal cord (cervical c 1)',\n",
       "  'coexpression Brain   Frontal Cortex (BA9)',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  '4DN dataset involves cell type',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Muscle   Skeletal',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'expresses',\n",
       "  'has signal',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Pancreas',\n",
       "  'correlated with condition',\n",
       "  'located in',\n",
       "  'coexpression Nerve   Tibial',\n",
       "  'coexpression Prostate',\n",
       "  'coexpression Vagina',\n",
       "  'coexpression Stomach',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'has parent',\n",
       "  'coexpression Colon   Transverse',\n",
       "  'coexpression Thyroid',\n",
       "  'gene associated with disease or phenotype',\n",
       "  '4DN loop ds start',\n",
       "  'located in',\n",
       "  'expressed in',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'located in',\n",
       "  'located in',\n",
       "  'correlated with condition',\n",
       "  'coexpression Brain   Nucleus accumbens (basal ganglia)',\n",
       "  '4DN loop us start',\n",
       "  'above 10kbp band',\n",
       "  'correlated with condition',\n",
       "  'has enzyme protein',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'expresses',\n",
       "  'coexpression Pituitary',\n",
       "  'coexpression Kidney   Cortex',\n",
       "  'coexpression Kidney   Medulla',\n",
       "  'correlated with condition',\n",
       "  'isa',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'coexpression Esophagus   Gastroesophageal Junction',\n",
       "  'coexpression Cervix   Ectocervix',\n",
       "  'expressed in',\n",
       "  'correlated with condition',\n",
       "  'indication'],\n",
       " 'selected': True,\n",
       " 'order': ['evidence', 'DESC'],\n",
       " 'color': '#bdbdbd',\n",
       " 'display': [{'label': 'source', 'text': '${source}', 'type': 'text'},\n",
       "  {'label': 'relation', 'text': '${relation}', 'type': 'text'},\n",
       "  {'label': 'target', 'text': '${target}', 'type': 'text'},\n",
       "  {'label': 'SAB', 'text': '${SAB}', 'type': 'text'},\n",
       "  {'label': 'evidence', 'text': '${evidence}', 'type': 'text'}]}"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edge_dict[edge_cols]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Additional schema\n",
    "\n",
    "The next part are miscellaneous schemas that you can use to modify our UI. Notice the tabs field, this is where we define the pages in our knowledge graph as well as the components that you want to render.\n",
    "\n",
    "Currently we have the following components available:\n",
    "1. KnowledgeGraph: this renders the single and two term search components\n",
    "2. Markdown: this is for rendering static markdown files, useful for tutorials\n",
    "3. Enrichment: For enrichment page.\n",
    "   \n",
    "Each component takes specific props like init_function that initializes/prefetch the static props for the page. For this example, we only look at the KnowledgeGraph component. For more information on how to use the other components you can check: https://maayanlab.cloud/enrichr-kg/api/knowledge_graph/schema"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "rest = {\n",
    "    \"header\": {\n",
    "\t\t\"title\": \"Data Distillery Knowledge Graph\",\n",
    "\t\t\"icon\": {\n",
    "\t\t\t\"src\": \"https://s3.amazonaws.com/maayan-kg/cfde-kg/assets/CFDE.png\",\n",
    "\t\t\t\"favicon\": \"https://s3.amazonaws.com/maayan-kg/cfde-kg/assets/favicon.png\",\n",
    "\t\t\t\"faviconTitle\": \"Data Distillery KG\",\n",
    "\t\t\t\"alt\": \"Data Distillery\",\n",
    "\t\t\t\"key\": \"data_distillery_logo\",\n",
    "\t\t\t\"width\": 100,\n",
    "\t\t\t\"height\": 100\n",
    "\t\t},\n",
    "\t\t\"tabs\": [\n",
    "\t\t\t{\n",
    "\t\t\t\t\"endpoint\": \"/\",\n",
    "\t\t\t\t\"label\": \"Term & Gene Search\",\n",
    "\t\t\t\t\"type\": \"page\",\n",
    "\t\t\t\t\"component\": \"KnowledgeGraph\",\n",
    "\t\t\t\t\"props\": {\n",
    "\t\t\t\t\t\"init_function\": \"initialize_kg\",\n",
    "                    \"initial_query\": {\n",
    "                        \"start\": \"Gene or Genome\",\n",
    "                        \"start_term\": \"CES1 gene\",\n",
    "                        \"start_field\": \"label\", \n",
    "                        \"relation\": \"positively regulated by\"\n",
    "\t\t\t\t\t}\n",
    "\t\t\t\t}\n",
    "\t\t\t}\n",
    "\t\t]\n",
    "\t},\n",
    "\t\"footer\": [\n",
    "\t\t{\n",
    "\t\t\t\"type\": \"github\",\n",
    "\t\t\t\"code\": \"https://github.com/MaayanLab/enrichr-kg\",\n",
    "\t\t\t\"issues\": \"https://github.com/MaayanLab/enrichr-kg/issues\"\n",
    "\t\t},\n",
    "\t\t{\n",
    "\t\t\t\"type\": \"icon\",\n",
    "\t\t\t\"key\": \"lab_icon\",\n",
    "\t\t\t\"src\": \"https://minio.dev.maayanlab.cloud/enrichr-kg/maayan-lab.jpg\",\n",
    "\t\t\t\"alt\": \"MaayanLab\",\n",
    "\t\t\t\"href\": \"https://labs.icahn.mssm.edu/maayanlab/\",\n",
    "\t\t\t\"height\": 125,\n",
    "\t\t\t\"width\": 150\n",
    "\t\t},\n",
    "\t\t{\n",
    "\t\t\t\"type\": \"icon\",\n",
    "\t\t\t\"key\": \"center_icon\",\n",
    "\t\t\t\"src\": \"https://minio.dev.maayanlab.cloud/enrichr-kg/ISMMS_Center_Bioinformatics_RGB_Vrtl.jpg\",\n",
    "\t\t\t\"alt\": \"COB\",\n",
    "\t\t\t\"href\": \"https://icahn.mssm.edu/research/bioinformatics\",\n",
    "\t\t\t\"height\": 125,\n",
    "\t\t\t\"width\": 200\n",
    "\t\t}\n",
    "\n",
    "\t]\n",
    "}"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Putting it all together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "schema = {\n",
    "    \"nodes\": node_schema,\n",
    "    \"edges\": edge_schema,\n",
    "    **rest\n",
    "}\n",
    "\n",
    "with open(\"../schema.json\", 'w') as o:\n",
    "    o.write(json.dumps(schema, indent=4))"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Once you have the schema, you can upload it and obtain a link, then set the following as env variable:\n",
    "\n",
    "```\n",
    "## landing endpoint\n",
    "NEXT_PUBLIC_PREFIX=\n",
    "\n",
    "## NEO4 specs\n",
    "NEO4J_URL=bolt://localhost:7687\n",
    "NEO4J_USER=neo4j\n",
    "NEO4J_NAME=kg\n",
    "NEO4J_PASSWORD=mysecretpassword\n",
    "\n",
    "# The link to the schema\n",
    "NEXT_PUBLIC_SCHEMA=https://s3.amazonaws.com/maayan-kg/enrichr-kg/assets/schema.json\n",
    "\n",
    "# set this if you are using assets (images etc) from different domains\n",
    "NEXT_PUBLIC_DOMAINS=minio.dokku.maayanlab.cloud,github.com,minio.dev.maayanlab.cloud,s3.amazonaws.com\n",
    "\n",
    "\n",
    "NEXT_PUBLIC_HOST=http://localhost:3000\n",
    "\n",
    "```\n",
    "\n",
    "Then run the following:\n",
    "\n",
    "```\n",
    "docker-compose up\n",
    "```\n",
    "\n",
    "http://localhost:3000/ should now be accessible (provided you kept the same prefix as above, if you set it to say, /kg, the the site is accessible on http://localhost:3000/kg)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "798122c9e4f43102ac6a3a7c51d7fe95ee01edcdc77ff31e968bccde04a8b5dc"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
