{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Building the UI\n",
    "In addition to the Neo4j console and the notebook earlier, we can provide develop a public facing website that we can share with the public without providing them write access to the neo4j database.\n",
    "\n",
    "The UI provides single and two-term search where the latter uses allShortestPath to query all the shortest paths between two terms. To build the UI, we need to create a UI schema that is built using the data that was ingested in the graph db. This notebook automates the creation of this schema file."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import re\n",
    "import json\n",
    "from glob import glob\n",
    "import pandas as pd\n",
    "import glasbey\n",
    "import seaborn as sns\n",
    "\n",
    "sns.set()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "115"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "counter = 0\n",
    "for filename in glob(\"../out/0915/filtered/nodes/*.nodes.csv\"):\n",
    "    counter += 1\n",
    "counter"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAIuQAAABhCAYAAAAY2jvwAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjcuMiwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy8pXeV/AAAACXBIWXMAAA9hAAAPYQGoP6dpAAAhL0lEQVR4nO3dW6id6Xkf8P/7ruM+ac+WNAd7pDhKK3AimGYEAoHbaTswMMUQhkCCYSimob4wGFIwvfBFwNTUJCQEQyBgKIGUkFJjcGMYCKE1U6YmQwdkFRWViZWpldm2ZXm2NNrHddyrF9NelnyLkn7srN/verP58/Ct9b3v8x5WWSwWiwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABAS2rbAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABYbbXtAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACsttp2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABWW207AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACrrbYdAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIDVVtsOAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMBqq20HAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGC11bYDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALDaatsBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFht3aZ/uFgs8pPZjzJfzP4m8/yt8vy4JJuXk8P3E3VrpnSTzZ9JniQ5bTvM2fFXG8nlfvL+JJkt2k5zNnRLcrm/yPuZZLZQtKaem0yzNtjMyfgwi1Mf0iZKrVkfbCYno8Sz1ti4s57+IJmMla2pUpL+YJH5zLhjGZNczlqvm5PpLIt42JooKVnrdXN0epiFAVtjm7Mk/XPJZD9ZqFsjpab0zyXjmBssYZRxBmuDjE/GWXiJNlJKyWBtkB9NzQ2WcXFxkPXBxRyPP8ipsUcjtXSzPriYJ5PjnHoXNHZ+Ok02d5LDx4l5aDO1flSzg3Ey973W1MHaWjYHyeHYo9ZUrcnmYJHjxfveBUsYjJ9Ob20z05NDD1tTtaa3tpnZyY+zWMzbTnNmHA0v56mafHhqStVUTfJUXWR2pCe5jDIYp9sbZDY1gW+uptsb5P35fma+1xq7lE563XOZzvaTqFszH9VsNp3rfi9hNj/N2rCXk9E0p14IjdRSsjbsZTTXk1zGYfc4T+d8fppHmXmHNtJNzdO5kNHMeG0Z/eleOuvnMz9+lBh7NFM6H9XsaOp7bQnT3mnWBsOcjEc51fNopNaatcEwB+OZVu4StmtS+t0sJjPvg4ZKSWq/m0x+ak/REhZlJ+kPk8nIPo+mSk36w+zPZzk1E22szo9zrr+d/ckT68gN1VKz3d9ODq21LGN/MMjWYJCD8Thzz1ojnVKzNRhkPH6cxan5VFPdspPusJPZaG681lApSXfYycF8L3Nz98Y6pZOtzoWM53tZWDdopKSTQedCDk5HxmtLqCnZqsPszkaZeYc20i01l7prOTlJTj1qjdWSrK8tktG+cW5TtWYxPJeTqbMtyyipWettZjo+1PNoqtT0BpuZHY98PpdRa3rrw+TE4l5jpWSx1s0Pc5Sp77XGeqm5lI3kaGHw0VQtWWyUzMY2zi+l1nQHg0zG72dhvaWRUrrpDy7n/cUiM/PQxrop+ZlSktFPE33JZmoni+HTOZg427KMWrrZ6l/O6ZG5QWOlpm5sZjyduJdiCSUlg14/e9NT74OGuim50KsZn5xmYYzbWKklg7VODmamBk3Vkmx1F5nOTvTXllBS0+uu5Xhy7LxBQ7WUrPfX8+D0OHM1a6xTSp6r65lNPrAHvKnSSbd/MePxxOdzCbWUDAb9nIwX9s43VErJ+qAkR3v25S6jdLPYuJD5+JGeR1O1k87gfBaH9jIvpdSUzWFO5gtzg4ZqSdY6JdNp9DyWUFLS6yWj6fs5jfdBEzXdDHuXMx4dGXcsoZSSwXAj4/FMr6ihUksGg26Op2N7JZdQU7LeG2Rv+iin9uU2UtPJhd75HM8+dB5oCbXUrHefyuPTH3vWGqrpZKd+LAezuZ7HEmop2ep28vh037PW0EfP2rkc6a8tpZaSjUE/k9mpcW5DpZT0uzVPxrFusIROKdkeJKfHerlNlVJS10uezA7jBrbmOinZ7m5lNkqUraGSdIeLzCdRs2WUpNNPxpMT32sNlVIy6K9lPj2Me1+XUdPpbeZ44tx7U6Uk6/1O5qOpnuQSSi3pDHs5HusVNVVLzfrgqSyOd53RWEIp3ZT1S3k81wNvqqZkpzPMqVfocmpSNxd5/3TPHQsNdUvN5Xoh4+leFmrWWCk1g96FzMfH5gYNlVLSGazndOy3apdSuqmDy3n/dOr3CRvqlpLLtZfT0bFDGsuoJXW4nrm7Y5qrNZ3huWQ8cS/FMkpJBv38eHqYuYFuI53UfKy3mcnkwHhtCaXU9PtbmY1nxmsNlVLSG3ST/bl36DLOdZNO+Wv/rNv0/5VS8i93/3neHf/3/6dcq+S/3nk6+ZVbyZ++lnzwvbbjnA0XX/yoZn+U5Edthzk7Xns9uXUtee1e8r3jttOcDS+uJ7eulbw2vpfvLRStqf90+y/y8s3X8/btb+fJwcO245wJ21vP5OWbrydv30qe7Lcd58y494l/kmvXk3t3k+OjttOcDesbybXrJXu7r2U6Nu5o6i867+Xlq1fy9v3dPBmN245zJmwPB3n56pV8Z/+N7M1/2nacM+Of/mSY7rXPZXbvG8nxg7bjnA3rz6V77XPJ7Vly0HaYs+Nu506uv3wjd9++k6Mnh23HORM2tjdz/eUb+Wfv3cudE3ODpv7N4t/m09d/L2/e/Vd5dPSXbcc5E85v/J18+vrv5Y/uvZkfHT9qO86Z8dW/3E0+86Xkja8nP32/7Thnw9OXP6rZN+8mPzGhaurb//gf5PWbybdvJw+NPRp5Zit5/WbJm4ev5dHcPLSpm+/8u1x95TO5/903MvrQnKqJ4VNP5+orn8nu25/P+MndtuOcGW/8o+/nizvJH+wnP3QevJHnO8kXd0p2vzPLeK/tNGfHxt+/nctXb+bB/dsZj7xEmxgMt3L56s289vCP873Jj9uOc2b8RedjuXr587n/4I8zGqtbE8PBRzXbvb+X8Wjadpwz4/39w7zyqU/mu7fey4f7J23HOROeOreWVz71ydx6eCf7Ez3Jpv7Dpf+cr3e/nN+Y/V6+n79qO86ZcDU/k693v5xbu8m+Zb3Grv/ga3n21a9k762vZfr4fttxzoTezify7KtfycPvvJfJnndBU9//+YO8cvOlfPf2O/nw4Enbcc6Ep7a288rNl/LNuw/ykyNfbE19YXuQrRcu5/DdB5mrWyOdjY9qlnu/kRx/v+04Z8Z88OV0X3gps3ffSY58rzWysZ3uCy/lG3u7eTD1+WzquSf/MZ+79oV8494f5cGxzfNNPLf+8Xzu2heSb7+RPLTW0tQ3/+GN/Nr1m/nm3dv5yZFebhPPbmzl167fzN3bv5+jA3P3pi6tfyVXPvVsdm/tZbyvL9nE4FwvVz71bL758Lfzk8n/bDvOmfFs/0p+7eO/lTt7v53Dqbo1sdm7khvP/la+tX8rD+fOtjT1TOdcPrvzqfzq7ju5PTY3aOIXB9v58ysv5btvJx8qWWNPbSevvFySd76RPHHeoJHt51Je+lzevv9GnozMDZraHj6dl69+JvfvfDujQ+dDmxhuPpOrN17PgzffyfiRL7amBue3c/nTLyVv7yZP9Ioa2R6kvHwln5n9Wf5bbGBr6u/lQv5L95eTPztM9myWbORCJ+WXt/Lg9u2MD/SJmhpsbeXyzZt5985rOTq0d76Jjc0X88KNW3nt5DDf8yPLjb1YO7m1vpW88+Xkyb2245wN23835aXfz7fuvZaHxz6fTT2z/mI+e+1W9r/z7cz3zA2a6Fx4Jju/9Hru3r+Xo5GzyE1tDNdz/eq1/Ob9x3lv5LLnJn5u2M3vXL2YO999nMMP1aypzae6ufHKxXxrN3loGtrIM4Pks1dK3nvw3ZyMP2w7zpmxNngqn7z8St589608Onrcdpwz4fzGTj79wqv5F4/fyv+YqVlTP9/dyb+/+Goe3PvXGR+/13acM2Gw/nO5fO1ruX33Xg6OjNea2tpYz83r1/L2ncM8OTR/b2J7s5OXb2wl3/nNZM/ns7ELP5fyS7+Tvdtfy/TA2ZYmelufyLM3v5KTP30npx9Yo2qqXtzO+q+8lO8+PMmHEz+Y0MRT/ZpXPr6e9+6PczJy8X9Ta8OST14d5J37r+XJSF+yie3hi3np6q3cufWdHO5bD21q89yF3PjUL+Xu7d0cHWh6NLGxNcj1m1fy5v27eTRy/1pT54cb+fTV6/nd+1/LD0bGa0387PAT+erVr+TN3T/Io/EP245zZpwfPJ9PX/livr7/+bw/d5dYE5c7v5Av7fxJvrX7MA/Hk7bjnBnPDPr57JWP5+v7f5z35+4pauJy52P50s7n82d33s3eoXdoUxc2N/LLN17Iu7uPczS2ttfExqCbF65czB/eHeeHR+ahTT2/UfLr1wd5/NZhZo/1cpvo7nRy8dWt/OGDP80Pxx+0HefMeH5wMb9++Vfy4FYydvyskcG55PKnSvbuzTK1RNVYbz159lo3d+69ncNjPfAmNte3c+Pay9m7/0amzlE11hs+nWevfiZvvbuXx0fOvTexs9HLqy88m4fvvJfJE/evNdXfXsvHX/pk3rz7B3l0pFfUxPmN5/Pp61/M4Vu/mvnj223HOTM6O7+YrVf/PF/fu5P3p+5+beJybzNfevZG9r+dzG2bb6zzTLLzeslrj38335v9oO04Z8KL3Z/NrYtfzZ33fjeHJz9oO86Zsbn2s7nxya/m4Z23Mjm0x7SJ/uZOPn7j1ezfeS1z5xwb62y+mJ0bt/La4/fyvZlxbhMvdtdy6+Ins//OW5k/8flsqrO9k52XXs3eO9/I1N0xjfS2n8uzL30us7vvJkfWqBrb2Ej3+gv5/P03clevqJFfGD6dP7n6mbx771s5OjY5aGpj/Zm8cO2z2b27m7F7wBsZbAxy5fqV5BsPkwf2eTT2heeTnd5f+2f1/0MUAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAPi/qm0HAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGC11bYDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALDaatsBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhtte0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKy22nYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFZbbTsAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKutth0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgNVW2w4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwGqrbQcAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAYLXVtgMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsNpq2wEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAWG217QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAArLbadgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAVlttOwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAq622HQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACA1VbbDgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAaqttBwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABgtdW2AwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACw2mrbAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABYbbXtAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACsttp2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABWW207AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACrrbYdAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIDVVtsOAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMBqq20HAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGC11bYDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALDaatsBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhtte0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKy22nYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFZbbTsAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKutth0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgNVW2w4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwGqrbQcAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAYLXVtgMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsNpq2wEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAWG217QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAArLbadgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAVlttOwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAq622HQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACA1VbbDgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAaqttBwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABgtdW2AwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACw2mrbAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABYbbXtAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACsttp2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABWW207AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACrrbYdAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIDVVtsOAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMBqq20HAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGC11bYDAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAALDaatsBAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFhtte0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKy22nYAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFZbbTsAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAKutth0AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAgNVW2w4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwGqrbQcAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAYLXVtgMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsNpq2wEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAWG217QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAArLbadgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAVlttOwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAq622HQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACA1VbbDgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAaqttBwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABgtdW2AwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACw2mrbAQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABYbbXtAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACsttp2AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABWW207AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACrrbYdAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAIDVVtsOAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAMBqq20HAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAGC1lcVisWj6xw+nP850MfmbzPO3yvOTTrJ5KTncTU7VrZHa/6hmT5LM2w5zduxuJpf6ye4kmZy2neZs6Nf/XbPFJJOFojX1sck0a8OtnIwOcnqqbk3UWrM23EpORomaNTbprqc/SCZjZWuq1qQ/SGbT3cR4rbFpuZy1Xi8n02lOmw+LV1otJWu9Xo7mhzk1YGtsc15S+ueymOwnp+rWSO2k9M8lo0XiXdDYuIwzWBtmfDLK6anvtSZqLRmsDfPjibnBMp7OYdYHF3M8/iDz02nbcc6ETu1lfXAxTybHmXsXNHZ+Nks2d5LDx8lc3RrpdD6q2cE4mflea+pgbS1bw+TA9L2xWpOtYXJ8upu5eWhjw/Ez6a1vZnp8mIX3QSOldj6q2cmDLIw7GjseXs5TneTDueWWpjpJnuok06NFTA2aq8Nxur1hZtNRFgrXSCk13d4wu7P9TBaztuOcGZdLN73uuUxn+1moWyPl/9RsOov2d3Oz09OsD/s5Hk301xqqtWR92M9oNrLWsoSj7ihPl538dPE40/hea6KXbp4uOxlNEx/P5vqzR+mun8/s+FFy6llrpHY/qtnRJAsPW2PT3mnWh2s5Hp3YU9RQrTXrw7UcjGeZedYa264lddDN6XiWJbY/r7RSPqpZJj9N9NcaW5SdlMFaFuOTaBY1VGrKYC37s2nm8flsqnN6knP97exPnlhHbqhTOznX304ODu3DWsLBcJCtwTAH41FmxmuNdGvN1mCY8ehxTs2nGuvWnfSG3UxHM6/QhkpNesNuDmZ7mel/N9Yt3Wx1L2Q028tCf62Rkm6G3Qs5mI8ytzG3sU5qtjrD/HB6Yo9pQ/1S83xvLccn9mEto9ZkfS3JifMGjdVOsnYuJ9PDnC7UrKlaOlnrbWY6OrDPo6FSanrDrcyOT7KYq1lTpVPTXV9LTizuNVZLstbLjxZHGdv119ggnXy8bCRHp8ncs9ZIpyQbNbPRKAsDtsZKrekOhxmPdrOwd76RUvoZDC9l9/Q0E+sGjfVTcqnW5OQDa3tN1V6ydjEHk93M3e3UWKf2s9W/lPnRgQl8U7Wms7GV8XSSU/OpxmqpGfT62ZvOM7XPo5FeKbnQ62R0PLd/bQmllgzXOzmYmho01SnJVi+ZzI71iZZQSk2/u57j8XHm6tZIp9SsD9bzk/lxpmrWWK/UPNtZz2zyQRb2LDRSajfd/sWMxhN7mZdQa81w0M/J6NSZoIZqKVkb1uRoz9x9GbWXbFzIbORsS2O1m+7wfE4PLb4vpdbUzbUcz04tUTVUS7LerZlMF85oLKGUkn6v5GS6m1PrBo3U0s9a71JGoyNrVEsotWY43Mh4NHXuvaFaSwbDXo6nY3P3JXRKzXpvkEfTR/YzN9Qt3Zzvnc/x9MPM7V9rrFM6We89lcfzB5nHnKqJTnrZ6TyXg+ksc+O1xjqlZKvXzeP5fubOGzTSSTc7nXM5Go09a0volJKN4SDj6dycqqFSSga9Tp6MF+5YWEK3lmwPSubHp9aRGyq1pLNe82R2mJm5QWPdUrPd3cxs5IqFpkpNusNkNvFbGkupSbdfMpqc6BU1VGrNsL+W2fQwMQ9trnTS7W3meDyzbtBQLcn6oJvZifvXllFqSXetn+Pxh+6OaahTO1kfPJXT4x9mYQ94Y6X2U9efz+PZyP1ODXVSstMdZn4Q47Vl1KSzlezOH7k/vaF+6eZS53xGk0fuTl9CKd0M++czG9nP3FQpNd3heuYjv1W7lNJPZ3gpu/NJJnq5jfRLyaVOP/OTY/vXllFrOmvrmbk7prnaSXftXBbjsXspllFLymCQB9PDTPWKGumVTp7rbWY8OfB7cUsotZNBfyvT8VSvqKFSS3qDXrI/c9BxGee6Hx14/GuUhZV5AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAFpU2w4AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAwGqrbQcAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAYLXVtgMAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAsNpq2wEAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAWG217QAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAArLbadgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAVlttOwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAq622HQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACA1VbbDgAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAADAavtfKYudIfA7oToAAAAASUVORK5CYII=",
      "text/plain": [
       "<Figure size 11500x100 with 1 Axes>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "palette = glasbey.create_palette(palette_size=counter, lightness_bounds=(65, 80))\n",
    "sns.palplot(palette)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'#35d71c'"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "palette[0]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Node Schema\n",
    "\n",
    "```\n",
    "{\n",
    "    \"node\": \"Gene\",\n",
    "    \"example\": [\n",
    "        \"COPB2\",\n",
    "        \"SDF2L1\"\n",
    "    ],\n",
    "    \"display\": [\n",
    "        {\n",
    "            \"label\": \"Gene ID\",\n",
    "            \"text\": \"${id}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"label\",\n",
    "            \"text\": \"${label}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"uri\",\n",
    "            \"text\": \"${uri}\",\n",
    "            \"href\": \"${uri}\",\n",
    "            \"type\": \"link\"\n",
    "        }\n",
    "    ],\n",
    "    \"search\": [\n",
    "        \"id\",\n",
    "        \"label\"\n",
    "    ],\n",
    "    \"color\": \"#c5e1a5\"\n",
    "}\n",
    "\n",
    "```\n",
    "This is a node schema for the node \"Gene\". The example field will display as an example on the UI while the display field tells the UI what to display as a tooltip. Meanwhile, search tells the UI that we can search both id and label fields of the nodes. Node colors are automatically generated but can otherwise be set using the field color for more customization."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "node_schema = []\n",
    "node_pattern = \"(?P<directory>.+)/(?P<label>.+)\\.(?P<entity>.+)\\.csv\"\n",
    "counter = 0\n",
    "for filename in glob(\"../out/0915/filtered/nodes/*.nodes.csv\"):\n",
    "    match = re.match(node_pattern, filename).groupdict()\n",
    "    label = match[\"label\"].replace(\"_\", \" \")\n",
    "    df = pd.read_csv(filename, low_memory=False)\n",
    "    display = []\n",
    "    search = []\n",
    "    for i in df.columns:\n",
    "        if i == \"id\":\n",
    "            display.append({\n",
    "                'label': '%s ID'%label,\n",
    "                'text': \"${%s}\"%i,\n",
    "                'type': 'text'\n",
    "            })\n",
    "        else:\n",
    "            val = str(df.at[0, i])\n",
    "            if val.startswith(\"http\"):\n",
    "                display.append({\n",
    "                    'label': i,\n",
    "                    'text': \"${%s}\"%i,\n",
    "                    'href': \"${%s}\"%i,\n",
    "                    'type': 'link'\n",
    "                })\n",
    "            else:\n",
    "                display.append({\n",
    "                    'label': i,\n",
    "                    'text': \"${%s}\"%i,\n",
    "                    'type': 'text'\n",
    "                })\n",
    "    examples = [i for i in df[\"label\"].unique() if type(i) == str][0:2]\n",
    "    node = {\n",
    "        \"node\": label,\n",
    "        \"example\": examples,\n",
    "        \"display\": display,\n",
    "        \"relation\": [],\n",
    "        \"search\": [i for i in df.columns if \"value\" not in i and \"lowerbound\" not in i and \"upperbound\" not in i ],\n",
    "        \"color\": palette[counter]\n",
    "    }\n",
    "    counter += 1\n",
    "    node_schema.append(node)\n",
    "        "
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Edge Schema\n",
    "```\n",
    "{\n",
    "    \"match\": [\n",
    "        \"KEGG\"\n",
    "    ],\n",
    "    \"selected\": true,\n",
    "    \"display\": [\n",
    "        {\n",
    "            \"label\": \"source\",\n",
    "            \"text\": \"${source}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"relation\",\n",
    "            \"text\": \"${relation}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"target\",\n",
    "            \"text\": \"${target}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"source label\",\n",
    "            \"text\": \"${source_label}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"target label\",\n",
    "            \"text\": \"${target_label}\",\n",
    "            \"type\": \"text\"\n",
    "        },\n",
    "        {\n",
    "            \"label\": \"resource\",\n",
    "            \"text\": \"${resource}\",\n",
    "            \"href\": \"${resource}\",\n",
    "            \"type\": \"link\"\n",
    "        }\n",
    "    ]\n",
    "}\n",
    "```\n",
    "\n",
    "This is the same with node schema with a few differences, first match allows you to match multiple edge types and selected is a field that allows us to select what edges are queried by default when doing two term searches."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edge_schema = []\n",
    "edge_pattern = \"(?P<directory>.+)/(?P<source_type>.+)\\.(?P<relation>.+)\\.(?P<target_type>.+)\\.(?P<entity>.+)\\.csv\"\n",
    "edge_dict = {}\n",
    "for filename in glob(\"../out/0915/filtered/edges/*.edges.csv\"):\n",
    "   match = re.match(edge_pattern, filename).groupdict()\n",
    "   entity = match[\"entity\"]\n",
    "   source_type = match[\"source_type\"]\n",
    "   relation = match[\"relation\"].replace(\"_\", \" \")\n",
    "   edge = {\n",
    "      # \"match\": [relation],\n",
    "      \"selected\": True,\n",
    "   }\n",
    "   order = [\"evidence\", \"ASC\"] if \"negative\" in relation.lower() else [\"evidence\", \"DESC\"]\n",
    "   color = \"#bdbdbd\"\n",
    "   if \"positive\" in relation.lower():\n",
    "       color = \"#90caf9\" \n",
    "   elif \"negative\" in relation.lower():\n",
    "      color = \"#ef9a9a\"\n",
    "   edge[\"order\"] = order\n",
    "   edge[\"color\"] = color\n",
    "   df = pd.read_csv(filename)\n",
    "   display = []\n",
    "   for i in df.columns:\n",
    "      if \"label\" not in i:\n",
    "         if i == \"id\":\n",
    "            display.append({\n",
    "                  'label': '%s ID'%label,\n",
    "                  'text': \"${%s}\"%i,\n",
    "                  'type': 'text'\n",
    "            })\n",
    "         else:\n",
    "            val = str(df.at[0, i])\n",
    "            if val.startswith(\"http\"):\n",
    "                  display.append({\n",
    "                     'label': i.replace(\"_\", \" \"),\n",
    "                     'text': \"${%s}\"%i,\n",
    "                     'href': \"${%s}\"%i,\n",
    "                     'type': 'link'\n",
    "                  })\n",
    "            else:\n",
    "                  display.append({\n",
    "                     'label': i.replace(\"_\", \" \"),\n",
    "                     'text': \"${%s}\"%i,\n",
    "                     'type': 'text'\n",
    "                  })\n",
    "            edge[\"display\"] = display\n",
    "   edge_cols = \",\".join(list(df.columns) + order + [color])\n",
    "   if edge_cols not in edge_dict:\n",
    "       edge_dict[edge_cols] = {\n",
    "           \"match\": [relation],\n",
    "           **edge\n",
    "       }\n",
    "   else:\n",
    "       edge_dict[edge_cols][\"match\"].append(relation)\n",
    "\n",
    "edge_schema = list(edge_dict.values())\n",
    "len(edge_schema)\n",
    "         "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'match': ['produces',\n",
       "  'associated with',\n",
       "  'expressed in',\n",
       "  'indication',\n",
       "  'regulates',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'expresses',\n",
       "  'causally influenced by',\n",
       "  'correlated with condition',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'expressed in',\n",
       "  'produced by',\n",
       "  'involved in',\n",
       "  'isa',\n",
       "  'indication',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'produced by',\n",
       "  'phenotype of',\n",
       "  'causally influences',\n",
       "  'has part',\n",
       "  'indication',\n",
       "  'molecularly interacts with',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'associated with',\n",
       "  'causally influences',\n",
       "  'located in',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'overlaps',\n",
       "  'coexpression Pancreas',\n",
       "  'is subsequence of',\n",
       "  'is approximately equivalent to',\n",
       "  'is approximately equivalent to',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'correlated with condition',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'signature gene of',\n",
       "  'synthesized by',\n",
       "  'indication',\n",
       "  'associated with',\n",
       "  'phenotype of',\n",
       "  'produced by',\n",
       "  'associated with',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'correlated with condition',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'correlated with condition',\n",
       "  'isa',\n",
       "  'produces',\n",
       "  'is approximately equivalent to',\n",
       "  'correlated with condition',\n",
       "  'bioactivity',\n",
       "  'coexpression Breast   Mammary Tissue',\n",
       "  'coexpression Adrenal Gland',\n",
       "  'has part',\n",
       "  'correlated with condition',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'has part',\n",
       "  'is approximately equivalent to',\n",
       "  'coexpression Minor Salivary Gland',\n",
       "  'causally influences',\n",
       "  'phenotype of',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'associated with',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'molecularly interacts with',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'causally influenced by',\n",
       "  'coexpression Heart   Left Ventricle',\n",
       "  'associated with',\n",
       "  'has part',\n",
       "  'correlated with condition',\n",
       "  'isa',\n",
       "  'causally influenced by',\n",
       "  'correlated with condition',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'produces',\n",
       "  'has phenotype',\n",
       "  'produced by',\n",
       "  'belongs to cohort',\n",
       "  'isa',\n",
       "  'coexpression Brain   Cerebellum',\n",
       "  'is approximately equivalent to',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'coexpression Kidney   Medulla',\n",
       "  'located in',\n",
       "  'produced by',\n",
       "  'coexpression Brain   Substantia nigra',\n",
       "  'correlated with condition',\n",
       "  'associated with',\n",
       "  'produces',\n",
       "  'indication',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'associated with',\n",
       "  'is approximately equivalent to',\n",
       "  'overlaps',\n",
       "  'coexpression Kidney   Cortex',\n",
       "  'coexpression Cells   Cultured fibroblasts',\n",
       "  'coexpression Colon   Transverse',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'involved in',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'phenotype of',\n",
       "  'causally influences',\n",
       "  'produced by',\n",
       "  'indication',\n",
       "  'phenotype of',\n",
       "  'produced by',\n",
       "  'indication',\n",
       "  'causally influenced by',\n",
       "  'isa',\n",
       "  'correlated with condition',\n",
       "  'refers to',\n",
       "  'produces',\n",
       "  'coexpression Brain   Hypothalamus',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'coexpression Brain   Spinal cord (cervical c 1)',\n",
       "  'correlated with condition',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'indication',\n",
       "  'coexpression Pituitary',\n",
       "  '4DN loop has qvalue bin',\n",
       "  'correlated with condition',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'location of',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'produced by',\n",
       "  'marker gene of',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'coexpression Adipose   Visceral (Omentum)',\n",
       "  'in similarity relationship with',\n",
       "  'is approximately equivalent to',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'produced by',\n",
       "  'isa',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'part of',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'regulates',\n",
       "  'has phenotype',\n",
       "  'produced by',\n",
       "  'is approximately equivalent to',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'has phenotype',\n",
       "  'correlated with condition',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'location of',\n",
       "  'associated with',\n",
       "  'produced by',\n",
       "  'has parent',\n",
       "  'causally influenced by',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  '4DN file has loop',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'coexpression Cervix   Endocervix',\n",
       "  'location of',\n",
       "  'produced by',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Fallopian Tube',\n",
       "  'produces',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'produces',\n",
       "  'has part',\n",
       "  'produces',\n",
       "  'coexpression Colon   Sigmoid',\n",
       "  'has phenotype',\n",
       "  'produced by',\n",
       "  'coexpression Spleen',\n",
       "  'not predicted in',\n",
       "  'has enzyme protein',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'causally influenced by',\n",
       "  'has phenotype',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'correlated with condition',\n",
       "  'coexpression Artery   Coronary',\n",
       "  'correlated with condition',\n",
       "  'produced by',\n",
       "  'attached by',\n",
       "  'causally influences',\n",
       "  'overlaps',\n",
       "  'correlated with condition',\n",
       "  'produces',\n",
       "  'coexpression Artery   Tibial',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'isa',\n",
       "  'produces',\n",
       "  'located in',\n",
       "  'correlated with condition',\n",
       "  'is from source',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'causally influenced by',\n",
       "  'produces',\n",
       "  'is approximately equivalent to',\n",
       "  'produced by',\n",
       "  'associated with',\n",
       "  'phenotype of',\n",
       "  'produced by',\n",
       "  'coexpression Skin   Not Sun Exposed (Suprapubic)',\n",
       "  'coexpression Esophagus   Muscularis',\n",
       "  'involved in',\n",
       "  'is approximately equivalent to',\n",
       "  'involved in',\n",
       "  'associated with',\n",
       "  'is approximately equivalent to',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'coexpression Cells   EBV transformed lymphocytes',\n",
       "  'involved in',\n",
       "  'has subsequence',\n",
       "  'coexpression Thyroid',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'is approximately equivalent to',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'has expression',\n",
       "  'has phenotype',\n",
       "  'produced by',\n",
       "  'has phenotype',\n",
       "  'produced by',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'coexpression Muscle   Skeletal',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'coexpression Artery   Aorta',\n",
       "  'belongs to cohort',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'causally influences',\n",
       "  'phenotype of',\n",
       "  'produced by',\n",
       "  'has phenotype',\n",
       "  'produces',\n",
       "  'isa',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'indication',\n",
       "  'phenotype of',\n",
       "  'associated with',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'is approximately equivalent to',\n",
       "  'causally influences',\n",
       "  'has canonical residue',\n",
       "  'produced by',\n",
       "  '4DN loop us end',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  '4DN loop ds end',\n",
       "  'correlated with condition',\n",
       "  'is subsequence of',\n",
       "  'produced by',\n",
       "  'coexpression Brain   Putamen (basal ganglia)',\n",
       "  'coexpression Brain   Anterior cingulate cortex (BA24)',\n",
       "  'isa',\n",
       "  'has phenotype',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'has enzyme protein',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'overlaps',\n",
       "  'produced by',\n",
       "  'correlated with condition',\n",
       "  'regulated by',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'associated with',\n",
       "  'molecularly interacts with',\n",
       "  'produced by',\n",
       "  'is subsequence of',\n",
       "  'gene associated with pathway',\n",
       "  'isa',\n",
       "  'location of',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'is approximately equivalent to',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'correlated with condition',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'overlaps',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'coexpression Brain   Cerebellar Hemisphere',\n",
       "  'located in',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'is approximately equivalent to',\n",
       "  'has glycosequence',\n",
       "  'regulated by',\n",
       "  'expressed in',\n",
       "  'produces',\n",
       "  'p value',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'has part',\n",
       "  'associated with',\n",
       "  'associated with',\n",
       "  'produced by',\n",
       "  'expresses',\n",
       "  'produces',\n",
       "  'is approximately equivalent to',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'involved in',\n",
       "  'produced by',\n",
       "  'is approximately equivalent to',\n",
       "  'causally influenced by',\n",
       "  'produces',\n",
       "  'location of',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'coexpression Heart   Atrial Appendage',\n",
       "  'coexpression Testis',\n",
       "  'correlated with condition',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'causally influenced by',\n",
       "  'causally influences',\n",
       "  'produced by',\n",
       "  'correlated with condition',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'associated with',\n",
       "  'correlated with condition',\n",
       "  '4DN dataset has file',\n",
       "  'part of',\n",
       "  'coexpression Brain   Nucleus accumbens (basal ganglia)',\n",
       "  'produced by',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'produced by',\n",
       "  'causally influences',\n",
       "  'indication',\n",
       "  'produced by',\n",
       "  'associated with',\n",
       "  'produced by',\n",
       "  'correlated with condition',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'produced by',\n",
       "  'predicted in',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Brain   Hippocampus',\n",
       "  'indication',\n",
       "  'associated with',\n",
       "  'indication',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'has subsequence',\n",
       "  'produces',\n",
       "  'overlaps',\n",
       "  'correlated in',\n",
       "  'indication',\n",
       "  'associated with',\n",
       "  'overlaps',\n",
       "  'associated with',\n",
       "  'location of',\n",
       "  'involved in',\n",
       "  'is approximately equivalent to',\n",
       "  'correlated with condition',\n",
       "  'associated with',\n",
       "  'produces',\n",
       "  'coexpression Stomach',\n",
       "  'involved in',\n",
       "  'produces',\n",
       "  'causally influenced by',\n",
       "  'correlated with condition',\n",
       "  'indication',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'gene located in chr band',\n",
       "  'indication',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'is approximately equivalent to',\n",
       "  'associated with',\n",
       "  'coexpression Liver',\n",
       "  'phenotype of',\n",
       "  'involved in',\n",
       "  'produced by',\n",
       "  'location of',\n",
       "  'coexpression Brain   Frontal Cortex (BA9)',\n",
       "  'location has variants',\n",
       "  'coexpression Brain   Amygdala',\n",
       "  'overlaps',\n",
       "  'associated with',\n",
       "  'is approximately equivalent to',\n",
       "  'regulates',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'produced by',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'produced by',\n",
       "  'causally influences',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'indication',\n",
       "  'indication',\n",
       "  'located in',\n",
       "  'associated with',\n",
       "  'produced by',\n",
       "  'coexpression Nerve   Tibial',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'has motif',\n",
       "  'produces',\n",
       "  'associated with',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'involved in',\n",
       "  'associated with',\n",
       "  'correlated with condition',\n",
       "  'coexpression Cervix   Ectocervix',\n",
       "  'causally influenced by',\n",
       "  'indication',\n",
       "  'causally influences',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'involved in',\n",
       "  'involved in',\n",
       "  'produces',\n",
       "  'has monosaccharide',\n",
       "  'is approximately equivalent to',\n",
       "  'indication',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'associated with',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'has phenotype',\n",
       "  'produces',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'coexpression Adipose   Subcutaneous',\n",
       "  'produces',\n",
       "  'causally influenced by',\n",
       "  'produces',\n",
       "  'causally influences',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'coexpression Esophagus   Gastroesophageal Junction',\n",
       "  'coexpression Uterus',\n",
       "  'produces',\n",
       "  'gene associated with disease or phenotype',\n",
       "  '4DN dataset involves cell type',\n",
       "  'part of',\n",
       "  'associated with',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'involved in',\n",
       "  'produces',\n",
       "  'coexpression Prostate',\n",
       "  'causally influences',\n",
       "  'associated with',\n",
       "  'part of',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'expresses',\n",
       "  'has signal',\n",
       "  'correlated with condition',\n",
       "  'located in',\n",
       "  'associated with',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'not correlated in',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'part of',\n",
       "  'correlated with condition',\n",
       "  'coexpression Vagina',\n",
       "  'phenotype of',\n",
       "  'is approximately equivalent to',\n",
       "  'produces',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'involved in',\n",
       "  'correlated with condition',\n",
       "  'coexpression Brain   Cortex',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'associated with',\n",
       "  'produced by',\n",
       "  'associated with',\n",
       "  'produces',\n",
       "  '4DN loop ds start',\n",
       "  'produces',\n",
       "  'located in',\n",
       "  'produces',\n",
       "  'coexpression Small Intestine   Terminal Ileum',\n",
       "  'coexpression Bladder',\n",
       "  'expressed in',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'coexpression Whole Blood',\n",
       "  'has phenotype',\n",
       "  'causally influenced by',\n",
       "  'produces',\n",
       "  'is approximately equivalent to',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'coexpression Esophagus   Mucosa',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'gender',\n",
       "  '4DN loop us start',\n",
       "  'produced by',\n",
       "  'is approximately equivalent to',\n",
       "  'is approximately equivalent to',\n",
       "  'produced by',\n",
       "  'has subsequence',\n",
       "  'phenotype of',\n",
       "  'produced by',\n",
       "  'gene associated with disease or phenotype',\n",
       "  'gene expression targeted by',\n",
       "  'correlated with condition',\n",
       "  'produced by',\n",
       "  'coexpression Skin   Sun Exposed (Lower leg)',\n",
       "  'produced by',\n",
       "  'coexpression Ovary',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'produces',\n",
       "  'causally influences',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'has phenotype',\n",
       "  'expresses',\n",
       "  'involved in',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'has phenotype',\n",
       "  'phenotype of',\n",
       "  'produces',\n",
       "  'correlated with condition',\n",
       "  'isa',\n",
       "  'phenotype of',\n",
       "  'located in',\n",
       "  'correlated with condition',\n",
       "  'involved in',\n",
       "  'indication',\n",
       "  'correlated with condition',\n",
       "  'is approximately equivalent to',\n",
       "  'produces',\n",
       "  'causally influenced by',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'produced by',\n",
       "  'associated with',\n",
       "  'regulated by',\n",
       "  'produces',\n",
       "  'produces',\n",
       "  'produced by',\n",
       "  'correlated with condition',\n",
       "  'indication',\n",
       "  'location of',\n",
       "  'involved in',\n",
       "  'coexpression Lung',\n",
       "  'produced by',\n",
       "  'coexpression Brain   Caudate (basal ganglia)',\n",
       "  'produced by',\n",
       "  'part of'],\n",
       " 'selected': True,\n",
       " 'order': ['evidence', 'DESC'],\n",
       " 'color': '#bdbdbd',\n",
       " 'display': [{'label': 'Unnamed: 0', 'text': '${Unnamed: 0}', 'type': 'text'},\n",
       "  {'label': 'source', 'text': '${source}', 'type': 'text'},\n",
       "  {'label': 'relation', 'text': '${relation}', 'type': 'text'},\n",
       "  {'label': 'target', 'text': '${target}', 'type': 'text'},\n",
       "  {'label': 'SAB', 'text': '${SAB}', 'type': 'text'},\n",
       "  {'label': 'evidence', 'text': '${evidence}', 'type': 'text'},\n",
       "  {'label': 'DCC', 'text': '${DCC}', 'type': 'text'}]}"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edge_dict[edge_cols]"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Additional schema\n",
    "\n",
    "The next part are miscellaneous schemas that you can use to modify our UI. Notice the tabs field, this is where we define the pages in our knowledge graph as well as the components that you want to render.\n",
    "\n",
    "Currently we have the following components available:\n",
    "1. KnowledgeGraph: this renders the single and two term search components\n",
    "2. Markdown: this is for rendering static markdown files, useful for tutorials\n",
    "3. Enrichment: For enrichment page.\n",
    "   \n",
    "Each component takes specific props like init_function that initializes/prefetch the static props for the page. For this example, we only look at the KnowledgeGraph component. For more information on how to use the other components you can check: https://maayanlab.cloud/enrichr-kg/api/knowledge_graph/schema"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "rest = {\n",
    "    \"header\": {\n",
    "\t\t\"title\": \"Data Distillery Knowledge Graph\",\n",
    "\t\t\"icon\": {\n",
    "\t\t\t\"src\": \"https://s3.amazonaws.com/maayan-kg/cfde-kg/assets/CFDE.png\",\n",
    "\t\t\t\"favicon\": \"https://s3.amazonaws.com/maayan-kg/cfde-kg/assets/favicon.png\",\n",
    "\t\t\t\"faviconTitle\": \"Data Distillery KG\",\n",
    "\t\t\t\"alt\": \"Data Distillery\",\n",
    "\t\t\t\"key\": \"data_distillery_logo\",\n",
    "\t\t\t\"width\": 100,\n",
    "\t\t\t\"height\": 100\n",
    "\t\t},\n",
    "\t\t\"tabs\": [\n",
    "\t\t\t{\n",
    "\t\t\t\t\"endpoint\": \"/\",\n",
    "\t\t\t\t\"label\": \"Term & Gene Search\",\n",
    "\t\t\t\t\"type\": \"page\",\n",
    "\t\t\t\t\"component\": \"KnowledgeGraph\",\n",
    "\t\t\t\t\"props\": {\n",
    "\t\t\t\t\t\"init_function\": \"initialize_kg\",\n",
    "                    \"initial_query\": {\n",
    "                        \"start\": \"Gene or Genome\",\n",
    "                        \"start_term\": \"CES1 gene\",\n",
    "                        \"start_field\": \"label\", \n",
    "                        \"relation\": \"positively regulated by\"\n",
    "\t\t\t\t\t}\n",
    "\t\t\t\t}\n",
    "\t\t\t}\n",
    "\t\t]\n",
    "\t},\n",
    "\t\"footer\": [\n",
    "\t\t{\n",
    "\t\t\t\"type\": \"github\",\n",
    "\t\t\t\"code\": \"https://github.com/MaayanLab/enrichr-kg\",\n",
    "\t\t\t\"issues\": \"https://github.com/MaayanLab/enrichr-kg/issues\"\n",
    "\t\t},\n",
    "\t\t{\n",
    "\t\t\t\"type\": \"icon\",\n",
    "\t\t\t\"key\": \"lab_icon\",\n",
    "\t\t\t\"src\": \"https://minio.dev.maayanlab.cloud/enrichr-kg/maayan-lab.jpg\",\n",
    "\t\t\t\"alt\": \"MaayanLab\",\n",
    "\t\t\t\"href\": \"https://labs.icahn.mssm.edu/maayanlab/\",\n",
    "\t\t\t\"height\": 125,\n",
    "\t\t\t\"width\": 150\n",
    "\t\t},\n",
    "\t\t{\n",
    "\t\t\t\"type\": \"icon\",\n",
    "\t\t\t\"key\": \"center_icon\",\n",
    "\t\t\t\"src\": \"https://minio.dev.maayanlab.cloud/enrichr-kg/ISMMS_Center_Bioinformatics_RGB_Vrtl.jpg\",\n",
    "\t\t\t\"alt\": \"COB\",\n",
    "\t\t\t\"href\": \"https://icahn.mssm.edu/research/bioinformatics\",\n",
    "\t\t\t\"height\": 125,\n",
    "\t\t\t\"width\": 200\n",
    "\t\t}\n",
    "\n",
    "\t]\n",
    "}"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Putting it all together"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "schema = {\n",
    "    \"nodes\": node_schema,\n",
    "    \"edges\": edge_schema,\n",
    "    **rest\n",
    "}\n",
    "\n",
    "with open(\"../schema.json\", 'w') as o:\n",
    "    o.write(json.dumps(schema, indent=4))"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Once you have the schema, you can upload it and obtain a link, then set the following as env variable:\n",
    "\n",
    "```\n",
    "## landing endpoint\n",
    "NEXT_PUBLIC_PREFIX=\n",
    "\n",
    "## NEO4 specs\n",
    "NEO4J_URL=bolt://localhost:7687\n",
    "NEO4J_USER=neo4j\n",
    "NEO4J_NAME=kg\n",
    "NEO4J_PASSWORD=mysecretpassword\n",
    "\n",
    "# The link to the schema\n",
    "NEXT_PUBLIC_SCHEMA=https://s3.amazonaws.com/maayan-kg/enrichr-kg/assets/schema.json\n",
    "\n",
    "# set this if you are using assets (images etc) from different domains\n",
    "NEXT_PUBLIC_DOMAINS=minio.dokku.maayanlab.cloud,github.com,minio.dev.maayanlab.cloud,s3.amazonaws.com\n",
    "\n",
    "\n",
    "NEXT_PUBLIC_HOST=http://localhost:3000\n",
    "\n",
    "```\n",
    "\n",
    "Then run the following:\n",
    "\n",
    "```\n",
    "docker-compose up\n",
    "```\n",
    "\n",
    "http://localhost:3000/ should now be accessible (provided you kept the same prefix as above, if you set it to say, /kg, the the site is accessible on http://localhost:3000/kg)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "venv",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.9"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "798122c9e4f43102ac6a3a7c51d7fe95ee01edcdc77ff31e968bccde04a8b5dc"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
